<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Finds overlapping junctions and annotates ERs as exonic, intronic, intergenic,
some combination or none of those — annotatERs • ODER</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Finds overlapping junctions and annotates ERs as exonic, intronic, intergenic,
some combination or none of those — annotatERs" />
<meta property="og:description" content="Looks at junctions passed in to find any overlaps and adds them in along with
other information as metadata columns. Then uses a gtf file or a Txdb passed
in to generate a genomic state and then labels each ER as to whether they are
exonic, intronic, intergenic on none" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ODER</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ODERflow.html">ODER workflow</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/eolagbaju/ODER/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Finds overlapping junctions and annotates ERs as exonic, intronic, intergenic,
some combination or none of those</h1>
    <small class="dont-index">Source: <a href='https://github.com/eolagbaju/ODER/blob/master/R/annotatER.R'><code>R/annotatER.R</code></a></small>
    <div class="hidden name"><code>annotatERs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Looks at junctions passed in to find any overlaps and adds them in along with
other information as metadata columns. Then uses a gtf file or a Txdb passed
in to generate a genomic state and then labels each ER as to whether they are
exonic, intronic, intergenic on none</p>
    </div>

    <pre class="usage"><span class='fu'>annotatERs</span><span class='op'>(</span>
  <span class='va'>opt_ers</span>,
  <span class='va'>junc_data</span>,
  <span class='va'>gtf_path</span>,
  txdb <span class='op'>=</span> <span class='cn'>NULL</span>,
  chrs_to_keep <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>22</span>, <span class='st'>"X"</span>, <span class='st'>"Y"</span>, <span class='st'>"MT"</span><span class='op'>)</span>,
  ensembl <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>opt_ers</th>
      <td><p>optimally defined ERs (the product of the ODER function)</p></td>
    </tr>
    <tr>
      <th>junc_data</th>
      <td><p>junction data that should match the ERs passed into opt_ers</p></td>
    </tr>
    <tr>
      <th>gtf_path</th>
      <td><p>a gtf file with exon data</p></td>
    </tr>
    <tr>
      <th>txdb</th>
      <td><p>txdb object, if one is not entered a gtf file needs to be</p></td>
    </tr>
    <tr>
      <th>chrs_to_keep</th>
      <td><p>chromosomes to keep in genomic state (in NCBI format i.e.
1,2,3...22,X,Y,MT)</p></td>
    </tr>
    <tr>
      <th>ensembl</th>
      <td><p>logical variable to say whether the gtf file entered is from
ensembl, the default is true</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>annotated ERs</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># \dontshow{</span>
<span class='va'>url</span> <span class='op'>&lt;-</span> <span class='fu'>recount</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/recount/man/download_study.html'>download_study</a></span><span class='op'>(</span>
    project <span class='op'>=</span> <span class='st'>"SRP012682"</span>,
    type <span class='op'>=</span> <span class='st'>"samples"</span>,
    download <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span> <span class='co'># .file_cache is an internal function to download a bigwig file from a link</span>
<span class='co'># if the file has been downloaded recently, it will be retrieved from a cache</span>

<span class='va'>bw_path</span> <span class='op'>&lt;-</span> <span class='fu'>ODER</span><span class='fu'>:::</span><span class='fu'>.file_cache</span><span class='op'>(</span><span class='va'>url</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>)</span>
<span class='va'>gtf_url</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span>
    <span class='st'>"http://ftp.ensembl.org/pub/release-103/gtf/"</span>,
    <span class='st'>"homo_sapiens/Homo_sapiens.GRCh38.103.chr.gtf.gz"</span>
<span class='op'>)</span>
<span class='va'>gtf_path</span> <span class='op'>&lt;-</span> <span class='fu'>ODER</span><span class='fu'>:::</span><span class='fu'>.file_cache</span><span class='op'>(</span><span class='va'>gtf_url</span><span class='op'>)</span>
<span class='co'># }</span>

<span class='va'>opt_ers</span> <span class='op'>&lt;-</span> <span class='fu'><a href='ODER.html'>ODER</a></span><span class='op'>(</span>
    bw_paths <span class='op'>=</span> <span class='va'>bw_path</span>, auc_raw <span class='op'>=</span> <span class='va'>auc_example</span>,
    auc_target <span class='op'>=</span> <span class='fl'>40e6</span> <span class='op'>*</span> <span class='fl'>100</span>, chrs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"chr21"</span>, <span class='st'>"chr22"</span><span class='op'>)</span>,
    genome <span class='op'>=</span> <span class='st'>"hg38"</span>, mccs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>5</span>, <span class='fl'>10</span><span class='op'>)</span>, mrgs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>10</span>, <span class='fl'>20</span><span class='op'>)</span>,
    gtf <span class='op'>=</span> <span class='va'>gtf_path</span>, ucsc_chr <span class='op'>=</span> <span class='cn'>TRUE</span>, ignore.strand <span class='op'>=</span> <span class='cn'>TRUE</span>,
    exons_no_overlap <span class='op'>=</span> <span class='cn'>NULL</span>, bw_chr <span class='op'>=</span> <span class='st'>"chr"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "2021-08-24 21:59:02 - Obtaining mean coverage across 1 samples"
#&gt; [1] "2021-08-24 21:59:02 - chr21"
#&gt; [1] "2021-08-24 21:59:02 - chr22"
#&gt; [1] "2021-08-24 21:59:03 - Generating ERs for chr21"
#&gt; [1] "2021-08-24 21:59:04 - Generating ERs for chr22"
#&gt; [1] "2021-08-24 21:59:05 - Loading in GTF..."
#&gt; [1] "2021-08-24 21:59:39 - Obtaining non-overlapping exons"
#&gt; [1] "2021-08-24 21:59:41 - Calculating delta for ERs..."
#&gt; [1] "2021-08-24 21:59:41 - Obtaining optimal set of ERs..."</div><div class='input'>
<span class='va'>junctions</span> <span class='op'>&lt;-</span> <span class='fu'>SummarizedExperiment</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html'>rowRanges</a></span><span class='op'>(</span><span class='fu'>dasper</span><span class='fu'>::</span><span class='va'><a href='https://rdrr.io/pkg/dasper/man/junctions_example.html'>junctions_example</a></span><span class='op'>)</span>
<span class='va'>annot_ers</span> <span class='op'>&lt;-</span> <span class='fu'>annotatERs</span><span class='op'>(</span>
    opt_ers <span class='op'>=</span> <span class='va'>opt_ers</span><span class='op'>[[</span><span class='st'>"opt_ers"</span><span class='op'>]</span><span class='op'>]</span>, junc_data <span class='op'>=</span> <span class='va'>junctions</span>,
    gtf_path <span class='op'>=</span> <span class='va'>gtf_path</span>, chrs_to_keep <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"21"</span>, <span class='st'>"22"</span><span class='op'>)</span>, ensembl <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "2021-08-24 21:59:41 - Obtaining co-ordinates of annotated exons and junctions from gtf/gff3..."
#&gt; [1] "2021-08-24 21:59:41 - Importing gtf/gff3 as a TxDb..."</div><div class='output co'>#&gt; <span class='message'>Import genomic features from the file as a GRanges object ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Prepare the 'metadata' data frame ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Make the TxDb object ... </span></div><div class='output co'>#&gt; <span class='warning'>Warning: The "phase" metadata column contains non-NA values for features of type</span>
#&gt; <span class='warning'>  stop_codon. This information was ignored.</span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; [1] "2021-08-24 22:00:58 - Getting junction annotation using overlapping exons..."
#&gt; [1] "2021-08-24 22:01:00 - Tidying junction annotation..."
#&gt; [1] "2021-08-24 22:01:00 - Deriving junction categories..."
#&gt; [1] "2021-08-24 22:01:05 - done!"
#&gt; [1] "2021-08-24 22:01:05 - Finding junctions overlapping ers..."
#&gt; [1] "2021-08-24 22:02:37 - Generating a genomic state..."</div><div class='output co'>#&gt; <span class='message'>Import genomic features from the file as a GRanges object ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Prepare the 'metadata' data frame ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Make the TxDb object ... </span></div><div class='output co'>#&gt; <span class='warning'>Warning: The "phase" metadata column contains non-NA values for features of type</span>
#&gt; <span class='warning'>  stop_codon. This information was ignored.</span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='warning'>Warning: The "phase" metadata column contains non-NA values for features of type</span>
#&gt; <span class='warning'>  stop_codon. This information was ignored.</span></div><div class='output co'>#&gt; <span class='message'>extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:1 mapping between keys and columns</span></div><div class='output co'>#&gt; [1] "2021-08-24 22:04:31 - Annotating the Expressed regions..."</div><div class='output co'>#&gt; <span class='message'>2021-08-24 22:04:31 annotateRegions: counting</span></div><div class='output co'>#&gt; <span class='message'>2021-08-24 22:04:31 annotateRegions: annotating</span></div><div class='output co'>#&gt; [1] "2021-08-24 22:05:28 - done!"</div><div class='input'>
<span class='va'>annot_ers</span>
</div><div class='output co'>#&gt; GRanges object with 15977 ranges and 5 metadata columns:
#&gt;           seqnames            ranges strand |
#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |
#&gt;       [1]    chr21   5032176-5032217      * |
#&gt;       [2]    chr21   5033408-5033425      * |
#&gt;       [3]    chr21   5034717-5034756      * |
#&gt;       [4]    chr21   5035188-5035189      * |
#&gt;       [5]    chr21   5036577-5036581      * |
#&gt;       ...      ...               ...    ... .
#&gt;   [15973]    chr22 50798996-50799149      * |
#&gt;   [15974]    chr22 50799209-50799284      * |
#&gt;   [15975]    chr22 50799669-50799688      * |
#&gt;   [15976]    chr22 50799717-50799744      * |
#&gt;   [15977]    chr22 50800460-50800587      * |
#&gt;                                                                                         grl
#&gt;                                                                               &lt;GRangesList&gt;
#&gt;       [1]                                                           chr21:5026423-5323718:+
#&gt;       [2]                                                           chr21:5026423-5323718:+
#&gt;       [3]                                                           chr21:5026423-5323718:+
#&gt;       [4]                                                           chr21:5026423-5323718:+
#&gt;       [5]                                                           chr21:5026423-5323718:+
#&gt;       ...                                                                               ...
#&gt;   [15973]     chr22:50363416-50804892:-,chr22:50460674-50807396:+,chr22:50770378-50805745:-
#&gt;   [15974]     chr22:50363416-50804892:-,chr22:50460674-50807396:+,chr22:50770378-50805745:-
#&gt;   [15975] chr22:50363416-50804892:-,chr22:50460674-50807396:+,chr22:50770378-50805745:-,...
#&gt;   [15976] chr22:50363416-50804892:-,chr22:50460674-50807396:+,chr22:50770378-50805745:-,...
#&gt;   [15977] chr22:50363416-50804892:-,chr22:50460674-50807396:+,chr22:50770378-50805745:-,...
#&gt;                     genes  annotation  og_index       gene_source
#&gt;           &lt;CharacterList&gt; &lt;character&gt; &lt;integer&gt;       &lt;character&gt;
#&gt;       [1] ENSG00000277117        exon         1 nearest gtf genes
#&gt;       [2] ENSG00000277117        exon         2 nearest gtf genes
#&gt;       [3] ENSG00000277117        exon         3 nearest gtf genes
#&gt;       [4] ENSG00000277117        exon         4 nearest gtf genes
#&gt;       [5] ENSG00000277117        exon         5 nearest gtf genes
#&gt;       ...             ...         ...       ...               ...
#&gt;   [15973] ENSG00000184319        exon     15973 nearest gtf genes
#&gt;   [15974] ENSG00000184319        exon     15974 nearest gtf genes
#&gt;   [15975] ENSG00000184319        exon     15975 nearest gtf genes
#&gt;   [15976] ENSG00000184319        exon     15976 nearest gtf genes
#&gt;   [15977] ENSG00000184319        exon     15977 nearest gtf genes
#&gt;   -------
#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Emmanuel Olagbaju.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


