<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Get junctions that overlap the optimally defined ERs — get_junctions • ODER</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Get junctions that overlap the optimally defined ERs — get_junctions" />
<meta property="og:description" content="The junctions will be added to the ERs passed in as a metadata column, a
GRangeslist with each GRanges element of the list corresponding to it's
associated ER. Any ERs that do not have an overlapping junction will have an
empty GRanges. The respective genes that each ER could be associated with will
also be passed in as a metadata column, a character list." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ODER</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ODERflow.html">ODER workflow</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/eolagbaju/ODER/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get junctions that overlap the optimally defined ERs</h1>
    <small class="dont-index">Source: <a href='https://github.com/eolagbaju/ODER/blob/master/R/annotatER.R'><code>R/annotatER.R</code></a></small>
    <div class="hidden name"><code>get_junctions.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The junctions will be added to the ERs passed in as a metadata column, a
GRangeslist with each GRanges element of the list corresponding to it's
associated ER. Any ERs that do not have an overlapping junction will have an
empty GRanges. The respective genes that each ER could be associated with will
also be passed in as a metadata column, a character list.</p>
    </div>

    <pre class="usage"><span class='fu'>get_junctions</span><span class='op'>(</span><span class='va'>opt_ers</span>, <span class='va'>junc_data</span>, <span class='va'>gtf_path</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>opt_ers</th>
      <td><p>optimally defined ERs (the product of the ODER function)</p></td>
    </tr>
    <tr>
      <th>junc_data</th>
      <td><p>junction data that should match the ERs passed into opt_ers</p></td>
    </tr>
    <tr>
      <th>gtf_path</th>
      <td><p>a gtf file with exon data</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>optimally defined ers annotated with junction and gene information</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Each Granges of the GRangeslist will have the metadata columns of "in_ref",
"gene_id_start", "tx_name_start", "exon_name_start", "strand_start",
"exon_width_start","gene_id_end", "tx_name_end", "exon_name_end", "strand_end",
"exon_width_end", "gene_id_junction", "strand_junction", "type" and "er_index
add the columns I'm adding</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>dasper::junction...</p>
<p>add more detail to description i.e. columns added in mcols by junction_annot
, gene id er etc. also talk about the empty granges for no overlaps</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># \dontshow{</span>
<span class='va'>gtf_url</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span>
    <span class='st'>"http://ftp.ensembl.org/pub/release-103/gtf/"</span>,
    <span class='st'>"homo_sapiens/Homo_sapiens.GRCh38.103.chr.gtf.gz"</span>
<span class='op'>)</span>
<span class='co'># .file_cache is an internal function to download a bigwig file from a link</span>
<span class='co'># if the file has been downloaded recently, it will be retrieved from a cache</span>
<span class='va'>gtf_path</span> <span class='op'>&lt;-</span> <span class='fu'>ODER</span><span class='fu'>:::</span><span class='fu'>.file_cache</span><span class='op'>(</span><span class='va'>gtf_url</span><span class='op'>)</span>
<span class='co'># }</span>

<span class='va'>example_ers</span> <span class='op'>&lt;-</span> <span class='fu'>GenomicRanges</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></span><span class='op'>(</span>
    seqnames <span class='op'>=</span> <span class='fu'>S4Vectors</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Rle-class.html'>Rle</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"chr21"</span>, <span class='st'>"chr22"</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>10</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>,
    ranges <span class='op'>=</span> <span class='fu'>IRanges</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html'>IRanges</a></span><span class='op'>(</span>
        start <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
            <span class='fl'>5026423</span>, <span class='fl'>24738</span>, <span class='fl'>5032218</span>, <span class='fl'>5033895</span>, <span class='fl'>17554</span>, <span class='fl'>50800446</span>,
            <span class='fl'>50800539</span>, <span class='fl'>16570</span>, <span class='fl'>50800790</span>, <span class='fl'>15005</span>, <span class='fl'>20312</span>
        <span class='op'>)</span>,
        end <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
            <span class='fl'>5323718</span>, <span class='fl'>24891</span>, <span class='fl'>5033407</span>, <span class='fl'>5033980</span>, <span class='fl'>17728</span>, <span class='fl'>50800910</span>,
            <span class='fl'>50800817</span>, <span class='fl'>16723</span>, <span class='fl'>50800910</span>, <span class='fl'>15038</span>, <span class='fl'>20582</span>
        <span class='op'>)</span>,
        names <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>letters</span>, <span class='fl'>11</span><span class='op'>)</span>
    <span class='op'>)</span>,
    strand <span class='op'>=</span> <span class='fu'>S4Vectors</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Rle-class.html'>Rle</a></span><span class='op'>(</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"+"</span>, <span class='st'>"-"</span><span class='op'>)</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>6</span>, <span class='fl'>5</span><span class='op'>)</span><span class='op'>)</span>,
    score <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>11</span>,
    GC <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>0</span>, length <span class='op'>=</span> <span class='fl'>11</span><span class='op'>)</span>
<span class='op'>)</span>

<span class='va'>example_junctions</span> <span class='op'>&lt;-</span> <span class='fu'>SummarizedExperiment</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html'>rowRanges</a></span><span class='op'>(</span><span class='fu'>dasper</span><span class='fu'>::</span><span class='va'><a href='https://rdrr.io/pkg/dasper/man/junctions_example.html'>junctions_example</a></span><span class='op'>)</span>

<span class='va'>example_er_juncs</span> <span class='op'>&lt;-</span> <span class='fu'>get_junctions</span><span class='op'>(</span>
    opt_ers <span class='op'>=</span> <span class='va'>example_ers</span>,
    junc_data <span class='op'>=</span> <span class='va'>example_junctions</span>,
    gtf_path <span class='op'>=</span> <span class='va'>gtf_path</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "2021-07-25 16:27:06 - Obtaining co-ordinates of annotated exons and junctions from gtf/gff3..."
#&gt; [1] "2021-07-25 16:27:06 - Importing gtf/gff3 as a TxDb..."</div><div class='output co'>#&gt; <span class='message'>Import genomic features from the file as a GRanges object ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Prepare the 'metadata' data frame ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Make the TxDb object ... </span></div><div class='output co'>#&gt; <span class='warning'>Warning: The "phase" metadata column contains non-NA values for features of type</span>
#&gt; <span class='warning'>  stop_codon. This information was ignored.</span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; [1] "2021-07-25 16:28:20 - Getting junction annotation using overlapping exons..."
#&gt; [1] "2021-07-25 16:28:22 - Tidying junction annotation..."
#&gt; [1] "2021-07-25 16:28:22 - Deriving junction categories..."
#&gt; [1] "2021-07-25 16:28:25 - done!"
#&gt; [1] "2021-07-25 16:28:25 - Finding junctions overlapping ers..."</div><div class='input'>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='va'>example_er_juncs</span><span class='op'>)</span>
</div><div class='output co'>#&gt; GRanges object with 11 ranges and 4 metadata columns:
#&gt;     seqnames            ranges strand |     score        GC
#&gt;        &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
#&gt;   a    chr21   5026423-5323718      + |         1       1.0
#&gt;   b    chr21       24738-24891      + |         2       0.9
#&gt;   c    chr21   5032218-5033407      + |         3       0.8
#&gt;   d    chr21   5033895-5033980      + |         4       0.7
#&gt;   e    chr21       17554-17728      + |         5       0.6
#&gt;   f    chr21 50800446-50800910      + |         6       0.5
#&gt;   g    chr21 50800539-50800817      - |         7       0.4
#&gt;   h    chr21       16570-16723      - |         8       0.3
#&gt;   i    chr21 50800790-50800910      - |         9       0.2
#&gt;   j    chr21       15005-15038      - |        10       0.1
#&gt;   k    chr22       20312-20582      - |        11       0.0
#&gt;                                                                             grl
#&gt;                                                                   &lt;GRangesList&gt;
#&gt;   a chr21:5026423-5323718:+,chr21:5028226-5032052:+,chr21:5032218-5033407:+,...
#&gt;   b                                                                            
#&gt;   c                             chr21:5026423-5323718:+,chr21:5032218-5033407:+
#&gt;   d                             chr21:5026423-5323718:+,chr21:5033895-5033980:+
#&gt;   e                                                                            
#&gt;   f                                                                            
#&gt;   g                                                                            
#&gt;   h                                                                            
#&gt;   i                                                                            
#&gt;   j                                                                            
#&gt;   k                                                                            
#&gt;               genes
#&gt;     &lt;CharacterList&gt;
#&gt;   a ENSG00000277117
#&gt;   b                
#&gt;   c ENSG00000277117
#&gt;   d                
#&gt;   e                
#&gt;   f                
#&gt;   g                
#&gt;   h                
#&gt;   i                
#&gt;   j                
#&gt;   k                
#&gt;   -------
#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Emmanuel Olagbaju.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


