<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Get the expressed genes — get_expressed_genes • ODER</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Get the expressed genes — get_expressed_genes" />
<meta property="og:description" content="Takes in a gtf file and a dataframe with expressed genes from a specific
tissue. Filters the gtf file for genes and keeps those that are present in the
tissue dataframe passed in." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ODER</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ODERflow.html">ODER workflow</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/eolagbaju/ODER/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get the expressed genes</h1>
    <small class="dont-index">Source: <a href='https://github.com/eolagbaju/ODER/blob/master/R/get_expressed_genes.R'><code>R/get_expressed_genes.R</code></a></small>
    <div class="hidden name"><code>get_expressed_genes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Takes in a gtf file and a dataframe with expressed genes from a specific
tissue. Filters the gtf file for genes and keeps those that are present in the
tissue dataframe passed in.</p>
    </div>

    <pre class="usage"><span class='fu'>get_expressed_genes</span><span class='op'>(</span><span class='va'>gtf_path</span>, species <span class='op'>=</span> <span class='st'>"Homo_sapiens"</span>, <span class='va'>tissue_df</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>gtf_path</th>
      <td><p>gtf file path</p></td>
    </tr>
    <tr>
      <th>species</th>
      <td><p>character string containing the species to filter for,
Homo sapiens is the default</p></td>
    </tr>
    <tr>
      <th>tissue_df</th>
      <td><p>dataframe containing the expressed genes for a particular
tissue</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>GRanges with the expressed genes for a specific tissue</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># \dontshow{</span>
<span class='va'>gtf_url</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span>
    <span class='st'>"http://ftp.ensembl.org/pub/release-103/gtf/"</span>,
    <span class='st'>"homo_sapiens/Homo_sapiens.GRCh38.103.chr.gtf.gz"</span>
<span class='op'>)</span>
<span class='co'># .file_cache is an internal function to download a bigwig file from a link</span>
<span class='co'># if the file has been downloaded recently, it will be retrieved from a cache</span>
<span class='va'>gtf_path</span> <span class='op'>&lt;-</span> <span class='fu'>ODER</span><span class='fu'>:::</span><span class='fu'>.file_cache</span><span class='op'>(</span><span class='va'>gtf_url</span><span class='op'>)</span>
<span class='co'># }</span>
<span class='va'>lung_tissue</span> <span class='op'>&lt;-</span> <span class='fu'><a href='get_tissue.html'>get_tissue</a></span><span class='op'>(</span>tissue <span class='op'>=</span> <span class='st'>"lung"</span><span class='op'>)</span>

<span class='va'>lung_exp_genes</span> <span class='op'>&lt;-</span> <span class='fu'>get_expressed_genes</span><span class='op'>(</span>
    gtf_path <span class='op'>=</span> <span class='va'>gtf_path</span>,
    tissue_df <span class='op'>=</span> <span class='va'>lung_tissue</span>
<span class='op'>)</span>

<span class='va'>lung_exp_genes</span>
</div><div class='output co'>#&gt; GRanges object with 21393 ranges and 22 metadata columns:
#&gt;           seqnames        ranges strand |         source     type     score
#&gt;              &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; |       &lt;factor&gt; &lt;factor&gt; &lt;numeric&gt;
#&gt;       [1]     chr1   14404-29570      - | havana             gene        NA
#&gt;       [2]     chr1  89295-133723      - | ensembl_havana     gene        NA
#&gt;       [3]     chr1 135141-135895      - | havana             gene        NA
#&gt;       [4]     chr1 139790-140339      - | havana             gene        NA
#&gt;       [5]     chr1 141474-173862      - | havana             gene        NA
#&gt;       ...      ...           ...    ... .            ...      ...       ...
#&gt;   [21389]     chrM   10760-12137      + |          insdc     gene        NA
#&gt;   [21390]     chrM   12337-14148      + |          insdc     gene        NA
#&gt;   [21391]     chrM   14149-14673      - |          insdc     gene        NA
#&gt;   [21392]     chrM   14674-14742      - |          insdc     gene        NA
#&gt;   [21393]     chrM   14747-15887      + |          insdc     gene        NA
#&gt;               phase         gene_id gene_version   gene_name    gene_source
#&gt;           &lt;integer&gt;     &lt;character&gt;  &lt;character&gt; &lt;character&gt;    &lt;character&gt;
#&gt;       [1]      &lt;NA&gt; ENSG00000227232            5      WASH7P         havana
#&gt;       [2]      &lt;NA&gt; ENSG00000238009            6  AL627309.1 ensembl_havana
#&gt;       [3]      &lt;NA&gt; ENSG00000268903            1  AL627309.6         havana
#&gt;       [4]      &lt;NA&gt; ENSG00000239906            1  AL627309.2         havana
#&gt;       [5]      &lt;NA&gt; ENSG00000241860            7  AL627309.5         havana
#&gt;       ...       ...             ...          ...         ...            ...
#&gt;   [21389]      &lt;NA&gt; ENSG00000198886            2      MT-ND4          insdc
#&gt;   [21390]      &lt;NA&gt; ENSG00000198786            2      MT-ND5          insdc
#&gt;   [21391]      &lt;NA&gt; ENSG00000198695            2      MT-ND6          insdc
#&gt;   [21392]      &lt;NA&gt; ENSG00000210194            1       MT-TE          insdc
#&gt;   [21393]      &lt;NA&gt; ENSG00000198727            2      MT-CYB          insdc
#&gt;                     gene_biotype transcript_id transcript_version
#&gt;                      &lt;character&gt;   &lt;character&gt;        &lt;character&gt;
#&gt;       [1] unprocessed_pseudogene          &lt;NA&gt;               &lt;NA&gt;
#&gt;       [2]                 lncRNA          &lt;NA&gt;               &lt;NA&gt;
#&gt;       [3]   processed_pseudogene          &lt;NA&gt;               &lt;NA&gt;
#&gt;       [4]                 lncRNA          &lt;NA&gt;               &lt;NA&gt;
#&gt;       [5]                 lncRNA          &lt;NA&gt;               &lt;NA&gt;
#&gt;       ...                    ...           ...                ...
#&gt;   [21389]         protein_coding          &lt;NA&gt;               &lt;NA&gt;
#&gt;   [21390]         protein_coding          &lt;NA&gt;               &lt;NA&gt;
#&gt;   [21391]         protein_coding          &lt;NA&gt;               &lt;NA&gt;
#&gt;   [21392]                Mt_tRNA          &lt;NA&gt;               &lt;NA&gt;
#&gt;   [21393]         protein_coding          &lt;NA&gt;               &lt;NA&gt;
#&gt;           transcript_name transcript_source transcript_biotype         tag
#&gt;               &lt;character&gt;       &lt;character&gt;        &lt;character&gt; &lt;character&gt;
#&gt;       [1]            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;
#&gt;       [2]            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;
#&gt;       [3]            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;
#&gt;       [4]            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;
#&gt;       [5]            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;
#&gt;       ...             ...               ...                ...         ...
#&gt;   [21389]            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;
#&gt;   [21390]            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;
#&gt;   [21391]            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;
#&gt;   [21392]            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;
#&gt;   [21393]            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;
#&gt;           transcript_support_level exon_number     exon_id exon_version
#&gt;                        &lt;character&gt; &lt;character&gt; &lt;character&gt;  &lt;character&gt;
#&gt;       [1]                     &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;         &lt;NA&gt;
#&gt;       [2]                     &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;         &lt;NA&gt;
#&gt;       [3]                     &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;         &lt;NA&gt;
#&gt;       [4]                     &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;         &lt;NA&gt;
#&gt;       [5]                     &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;         &lt;NA&gt;
#&gt;       ...                      ...         ...         ...          ...
#&gt;   [21389]                     &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;         &lt;NA&gt;
#&gt;   [21390]                     &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;         &lt;NA&gt;
#&gt;   [21391]                     &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;         &lt;NA&gt;
#&gt;   [21392]                     &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;         &lt;NA&gt;
#&gt;   [21393]                     &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;         &lt;NA&gt;
#&gt;            protein_id protein_version     ccds_id
#&gt;           &lt;character&gt;     &lt;character&gt; &lt;character&gt;
#&gt;       [1]        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;
#&gt;       [2]        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;
#&gt;       [3]        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;
#&gt;       [4]        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;
#&gt;       [5]        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;
#&gt;       ...         ...             ...         ...
#&gt;   [21389]        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;
#&gt;   [21390]        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;
#&gt;   [21391]        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;
#&gt;   [21392]        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;
#&gt;   [21393]        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;
#&gt;   -------
#&gt;   seqinfo: 25 sequences from an unspecified genome; no seqlengths</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Emmanuel Olagbaju.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


