<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Get junctions that overlap the optimally defined ERs — get_junctions • ODER</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Get junctions that overlap the optimally defined ERs — get_junctions" />
<meta property="og:description" content="The junctions will be added to the ERs passed in as a metadata column, a
GRangeslist with each GRanges element of the list corresponding to it's
associated ER. If there are no overlaps, the GRangeslist will be empty.
Any ERs that do not have an overlapping junction will have an empty GRanges.
The respective genes that each ER could be associated with will also be
passed in as a metadata column, a character list." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ODER</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.18</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ODER_overview.html">ODER: Optimising the Definition of Expressed Regions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/eolagbaju/ODER/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get junctions that overlap the optimally defined ERs</h1>
    <small class="dont-index">Source: <a href='https://github.com/eolagbaju/ODER/blob/master/R/annotatER.R'><code>R/annotatER.R</code></a></small>
    <div class="hidden name"><code>get_junctions.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The junctions will be added to the ERs passed in as a metadata column, a
<code>GRangeslist</code> with each GRanges element of the list corresponding to it's
associated ER. If there are no overlaps, the <code>GRangeslist</code> will be empty.
Any ERs that do not have an overlapping junction will have an empty GRanges.
The respective genes that each ER could be associated with will also be
passed in as a metadata column, a character list.</p>
    </div>

    <div class="ref-usage sourceCode"><pre class='sourceCode r'><code><span class='fu'>get_junctions</span><span class='op'>(</span><span class='va'>opt_ers</span>, <span class='va'>junc_data</span>, <span class='va'>txdb</span><span class='op'>)</span></code></pre></div>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>opt_ers</th>
      <td><p>optimally defined ERs (the product of the ODER function)</p></td>
    </tr>
    <tr>
      <th>junc_data</th>
      <td><p>junction data that should match the ERs passed into opt_ers</p></td>
    </tr>
    <tr>
      <th>txdb</th>
      <td><p>a gtf file or txdb (<a href='https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html'>TxDb-class</a>)
with exon data. This will ultimately be used to annotate the junction data
passed in.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>optimally defined ers annotated with junction and gene information</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>dasper::junction_annot</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <div class="ref-examples sourceCode"><pre class='sourceCode r'><code><span class='r-in'><span class='va'>gtf_url</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span></span>
<span class='r-in'>    <span class='st'>"http://ftp.ensembl.org/pub/release-103/gtf/"</span>,</span>
<span class='r-in'>    <span class='st'>"homo_sapiens/Homo_sapiens.GRCh38.103.chr.gtf.gz"</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-in'><span class='co'># file_cache is an internal function to download a bigwig file from a link</span></span>
<span class='r-in'><span class='co'># if the file has been downloaded recently, it will be retrieved from a cache</span></span>
<span class='r-in'><span class='va'>gtf_path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='file_cache.html'>file_cache</a></span><span class='op'>(</span><span class='va'>gtf_url</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>example_ers</span> <span class='op'>&lt;-</span> <span class='fu'>GenomicRanges</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></span><span class='op'>(</span></span>
<span class='r-in'>    seqnames <span class='op'>=</span> <span class='fu'>S4Vectors</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Rle-class.html'>Rle</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"chr21"</span>, <span class='st'>"chr22"</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>10</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span class='r-in'>    ranges <span class='op'>=</span> <span class='fu'>IRanges</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html'>IRanges</a></span><span class='op'>(</span></span>
<span class='r-in'>        start <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span></span>
<span class='r-in'>            <span class='fl'>5026423</span>, <span class='fl'>24738</span>, <span class='fl'>5032218</span>, <span class='fl'>5033895</span>, <span class='fl'>17554</span>, <span class='fl'>50800446</span>,</span>
<span class='r-in'>            <span class='fl'>50800539</span>, <span class='fl'>16570</span>, <span class='fl'>50800790</span>, <span class='fl'>15005</span>, <span class='fl'>20312</span></span>
<span class='r-in'>        <span class='op'>)</span>,</span>
<span class='r-in'>        end <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span></span>
<span class='r-in'>            <span class='fl'>5323718</span>, <span class='fl'>24891</span>, <span class='fl'>5033407</span>, <span class='fl'>5033980</span>, <span class='fl'>17728</span>, <span class='fl'>50800910</span>,</span>
<span class='r-in'>            <span class='fl'>50800817</span>, <span class='fl'>16723</span>, <span class='fl'>50800910</span>, <span class='fl'>15038</span>, <span class='fl'>20582</span></span>
<span class='r-in'>        <span class='op'>)</span>,</span>
<span class='r-in'>        names <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>letters</span>, <span class='fl'>11</span><span class='op'>)</span></span>
<span class='r-in'>    <span class='op'>)</span>,</span>
<span class='r-in'>    strand <span class='op'>=</span> <span class='fu'>S4Vectors</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Rle-class.html'>Rle</a></span><span class='op'>(</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"+"</span>, <span class='st'>"-"</span><span class='op'>)</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>6</span>, <span class='fl'>5</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span class='r-in'>    score <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>11</span>,</span>
<span class='r-in'>    GC <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>0</span>, length <span class='op'>=</span> <span class='fl'>11</span><span class='op'>)</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-in'><span class='kw'>if</span> <span class='op'>(</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/exists.html'>exists</a></span><span class='op'>(</span><span class='st'>"gtf_gr"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span></span>
<span class='r-in'>    <span class='va'>gtf_gr</span> <span class='op'>&lt;-</span> <span class='fu'>rtracklayer</span><span class='fu'>::</span><span class='fu'>import</span><span class='op'>(</span><span class='va'>gtf_path</span><span class='op'>)</span></span>
<span class='r-in'><span class='op'>}</span></span>
<span class='r-in'><span class='va'>chrs_to_keep</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"21"</span>, <span class='st'>"22"</span><span class='op'>)</span></span>
<span class='r-in'><span class='co'>#### preparing the txdb and genomstate object(s)</span></span>
<span class='r-in'><span class='va'>hg38_chrominfo</span> <span class='op'>&lt;-</span> <span class='fu'>GenomeInfoDb</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/GenomeInfoDb/man/getChromInfoFromUCSC.html'>getChromInfoFromUCSC</a></span><span class='op'>(</span><span class='st'>"hg38"</span><span class='op'>)</span></span>
<span class='r-in'><span class='va'>new_info</span> <span class='op'>&lt;-</span> <span class='va'>hg38_chrominfo</span><span class='op'>$</span><span class='va'>size</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/match.html'>match</a></span><span class='op'>(</span></span>
<span class='r-in'>    <span class='va'>chrs_to_keep</span>,</span>
<span class='r-in'>    <span class='fu'>GenomeInfoDb</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html'>mapSeqlevels</a></span><span class='op'>(</span><span class='va'>hg38_chrominfo</span><span class='op'>$</span><span class='va'>chrom</span>, <span class='st'>"Ensembl"</span><span class='op'>)</span></span>
<span class='r-in'><span class='op'>)</span><span class='op'>]</span></span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>new_info</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>chrs_to_keep</span></span>
<span class='r-in'><span class='va'>gtf_gr_tx</span> <span class='op'>&lt;-</span> <span class='fu'>GenomeInfoDb</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/GenomeInfoDb/man/seqlevels-wrappers.html'>keepSeqlevels</a></span><span class='op'>(</span><span class='va'>gtf_gr</span>,</span>
<span class='r-in'>    <span class='va'>chrs_to_keep</span>,</span>
<span class='r-in'>    pruning.mode <span class='op'>=</span> <span class='st'>"tidy"</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-in'><span class='fu'>GenomeInfoDb</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html'>seqlengths</a></span><span class='op'>(</span><span class='va'>gtf_gr_tx</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>new_info</span></span>
<span class='r-in'><span class='fu'>GenomeInfoDb</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html'>seqlevelsStyle</a></span><span class='op'>(</span><span class='va'>gtf_gr_tx</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"UCSC"</span></span>
<span class='r-in'><span class='fu'>GenomeInfoDb</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html'>genome</a></span><span class='op'>(</span><span class='va'>gtf_gr_tx</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"hg38"</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>ucsc_txdb</span> <span class='op'>&lt;-</span> <span class='fu'>GenomicFeatures</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/GenomicFeatures/man/makeTxDbFromGRanges.html'>makeTxDbFromGRanges</a></span><span class='op'>(</span><span class='va'>gtf_gr_tx</span><span class='op'>)</span></span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span> <span class='warning'>Warning: </span>The "phase" metadata column contains non-NA values for features of type</span>
<span class='r-wrn co'><span class='r-pr'>#&gt;</span>   stop_codon. This information was ignored.</span>
<span class='r-in'><span class='va'>genom_state</span> <span class='op'>&lt;-</span> <span class='fu'>derfinder</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/derfinder/man/makeGenomicState.html'>makeGenomicState</a></span><span class='op'>(</span>txdb <span class='op'>=</span> <span class='va'>ucsc_txdb</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 'select()' returned 1:1 mapping between keys and columns</span>
<span class='r-in'><span class='va'>ens_txdb</span> <span class='op'>&lt;-</span> <span class='va'>ucsc_txdb</span></span>
<span class='r-in'><span class='fu'>GenomeInfoDb</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html'>seqlevelsStyle</a></span><span class='op'>(</span><span class='va'>ens_txdb</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"Ensembl"</span></span>
<span class='r-in'><span class='co'>################### end of genomstate creation</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>junctions</span> <span class='op'>&lt;-</span> <span class='fu'>SummarizedExperiment</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html'>rowRanges</a></span><span class='op'>(</span><span class='fu'>dasper</span><span class='fu'>::</span><span class='va'><a href='https://rdrr.io/pkg/dasper/man/junctions_example.html'>junctions_example</a></span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>example_er_juncs</span> <span class='op'>&lt;-</span> <span class='fu'>get_junctions</span><span class='op'>(</span></span>
<span class='r-in'>    opt_ers <span class='op'>=</span> <span class='va'>example_ers</span>,</span>
<span class='r-in'>    junc_data <span class='op'>=</span> <span class='va'>junctions</span>,</span>
<span class='r-in'>    txdb <span class='op'>=</span> <span class='va'>ens_txdb</span> <span class='co'># can either be a gtf file or txdb in the Ensembl format</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-29 14:21:41 - Obtaining co-ordinates of annotated exons and junctions..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-29 14:21:42 - Getting junction annotation using overlapping exons..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-29 14:21:44 - Tidying junction annotation..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-29 14:21:44 - Deriving junction categories..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-29 14:21:49 - done!"</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-09-29 14:21:49 - Finding junctions overlapping ers...</span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>example_er_juncs</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> GRanges object with 11 ranges and 4 metadata columns:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>     seqnames            ranges strand |     score        GC</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>        &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   a    chr21   5026423-5323718      + |         1       1.0</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   b    chr21       24738-24891      + |         2       0.9</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   c    chr21   5032218-5033407      + |         3       0.8</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   d    chr21   5033895-5033980      + |         4       0.7</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   e    chr21       17554-17728      + |         5       0.6</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   f    chr21 50800446-50800910      + |         6       0.5</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   g    chr21 50800539-50800817      - |         7       0.4</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   h    chr21       16570-16723      - |         8       0.3</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   i    chr21 50800790-50800910      - |         9       0.2</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   j    chr21       15005-15038      - |        10       0.1</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   k    chr22       20312-20582      - |        11       0.0</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                                                                             grl</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>                                                                   &lt;GRangesList&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   a chr21:5026423-5323718:+,chr21:5028226-5032052:+,chr21:5032218-5033407:+,...</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   b                                                                            </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   c                             chr21:5026423-5323718:+,chr21:5032218-5033407:+</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   d                             chr21:5026423-5323718:+,chr21:5033895-5033980:+</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   e                                                                            </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   f                                                                            </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   g                                                                            </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   h                                                                            </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   i                                                                            </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   j                                                                            </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   k                                                                            </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>               genes</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>     &lt;CharacterList&gt;</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   a ENSG00000277117</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   b                </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   c ENSG00000277117</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   d                </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   e                </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   f                </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   g                </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   h                </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   i                </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   j                </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   k                </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   -------</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
</code></pre></div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by Emmanuel Olagbaju.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


