<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Generate the count matrix — get_count_matrix • ODER</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Generate the count matrix — get_count_matrix" />
<meta property="og:description" content="Scores the mean coverage of the expressed regions as a count matrix" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ODER</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ODERflow.html">ODER workflow</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/eolagbaju/ODER/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate the count matrix</h1>
    <small class="dont-index">Source: <a href='https://github.com/eolagbaju/ODER/blob/master/R/get_count_matrix.R'><code>R/get_count_matrix.R</code></a></small>
    <div class="hidden name"><code>get_count_matrix.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Scores the mean coverage of the expressed regions as a count matrix</p>
    </div>

    <pre class="usage"><span class='fu'>get_count_matrix</span><span class='op'>(</span><span class='va'>bw_paths</span>, <span class='va'>annot_ers</span>, cols <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>bw_paths</th>
      <td><p>Vector containing the bigwig file paths to read in</p></td>
    </tr>
    <tr>
      <th>annot_ers</th>
      <td><p>GRangesList containing the annotated ERs (product of annotatERs())</p></td>
    </tr>
    <tr>
      <th>cols</th>
      <td><p>A dataframe containing the information to be used as colData for the
output. If NULL then the bw_paths will be used for the colData</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A Ranged Summarized Experiment containing the gene counts as an assay</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># \dontshow{</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://magrittr.tidyverse.org'>magrittr</a></span><span class='op'>)</span>
<span class='va'>gtex_metadata</span> <span class='op'>&lt;-</span> <span class='fu'>recount</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/recount/man/all_metadata.html'>all_metadata</a></span><span class='op'>(</span><span class='st'>"gtex"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>2021-08-24 17:24:32 downloading the metadata to /tmp/RtmpKiYpcx/metadata_clean_gtex.Rdata</span></div><div class='input'><span class='va'>gtex_metadata</span> <span class='op'>&lt;-</span> <span class='va'>gtex_metadata</span> <span class='op'>%&gt;%</span>
    <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
    <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>project</span> <span class='op'>==</span> <span class='st'>"SRP012682"</span><span class='op'>)</span>

<span class='va'>gtf_url</span> <span class='op'>&lt;-</span> <span class='st'>"http://ftp.ensembl.org/pub/release-103/gtf/homo_sapiens/Homo_sapiens.GRCh38.103.chr.gtf.gz"</span>
<span class='va'>gtf_path</span> <span class='op'>&lt;-</span> <span class='fu'>ODER</span><span class='fu'>:::</span><span class='fu'>.file_cache</span><span class='op'>(</span><span class='va'>gtf_url</span><span class='op'>)</span>

<span class='va'>url</span> <span class='op'>&lt;-</span> <span class='fu'>recount</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/recount/man/download_study.html'>download_study</a></span><span class='op'>(</span>
    project <span class='op'>=</span> <span class='st'>"SRP012682"</span>,
    type <span class='op'>=</span> <span class='st'>"samples"</span>,
    download <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span> <span class='co'># .file_cache is an internal function to download a bigwig file from a link</span>
<span class='co'># if the file has been downloaded recently, it will be retrieved from a cache</span>

<span class='va'>bw_path</span> <span class='op'>&lt;-</span> <span class='fu'>ODER</span><span class='fu'>:::</span><span class='fu'>.file_cache</span><span class='op'>(</span><span class='va'>url</span><span class='op'>[</span><span class='fl'>84</span><span class='op'>]</span><span class='op'>)</span>
<span class='va'>bw_path2</span> <span class='op'>&lt;-</span> <span class='fu'>ODER</span><span class='fu'>:::</span><span class='fu'>.file_cache</span><span class='op'>(</span><span class='va'>url</span><span class='op'>[</span><span class='fl'>91</span><span class='op'>]</span><span class='op'>)</span>

<span class='va'>opt_ers</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/warning.html'>suppressWarnings</a></span><span class='op'>(</span><span class='fu'><a href='ODER.html'>ODER</a></span><span class='op'>(</span>
    bw_paths <span class='op'>=</span> <span class='va'>bw_path</span>, auc_raw <span class='op'>=</span> <span class='va'>gtex_metadata</span><span class='op'>[[</span><span class='st'>"auc"</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span><span class='fl'>84</span><span class='op'>]</span>,
    auc_target <span class='op'>=</span> <span class='fl'>40e6</span> <span class='op'>*</span> <span class='fl'>100</span>, chrs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"chr21"</span>, <span class='st'>"chr22"</span><span class='op'>)</span>,
    genome <span class='op'>=</span> <span class='st'>"hg38"</span>, mccs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>2</span>, <span class='fl'>4</span>, <span class='fl'>6</span>, <span class='fl'>8</span>, <span class='fl'>10</span><span class='op'>)</span>, mrgs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>10</span>, <span class='fl'>20</span>, <span class='fl'>30</span><span class='op'>)</span>,
    gtf <span class='op'>=</span> <span class='va'>gtf_path</span>, ucsc_chr <span class='op'>=</span> <span class='cn'>TRUE</span>, ignore.strand <span class='op'>=</span> <span class='cn'>TRUE</span>,
    exons_no_overlap <span class='op'>=</span> <span class='cn'>NULL</span>, bw_chr <span class='op'>=</span> <span class='st'>"chr"</span>
<span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "2021-08-24 17:24:34 - Obtaining mean coverage across 1 samples"
#&gt; [1] "2021-08-24 17:24:34 - chr21"
#&gt; [1] "2021-08-24 17:24:34 - chr22"
#&gt; [1] "2021-08-24 17:24:34 - Generating ERs for chr21"
#&gt; [1] "2021-08-24 17:24:37 - Generating ERs for chr22"
#&gt; [1] "2021-08-24 17:24:39 - Loading in GTF..."
#&gt; [1] "2021-08-24 17:25:15 - Obtaining non-overlapping exons"
#&gt; [1] "2021-08-24 17:25:19 - Calculating delta for ERs..."
#&gt; [1] "2021-08-24 17:25:20 - Obtaining optimal set of ERs..."</div><div class='input'>
<span class='va'>opt_ers2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/warning.html'>suppressWarnings</a></span><span class='op'>(</span><span class='fu'><a href='ODER.html'>ODER</a></span><span class='op'>(</span>
    bw_paths <span class='op'>=</span> <span class='va'>bw_path</span>, auc_raw <span class='op'>=</span> <span class='va'>gtex_metadata</span><span class='op'>[[</span><span class='st'>"auc"</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span><span class='fl'>91</span><span class='op'>]</span>,
    auc_target <span class='op'>=</span> <span class='fl'>40e6</span> <span class='op'>*</span> <span class='fl'>100</span>, chrs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"chr21"</span>, <span class='st'>"chr22"</span><span class='op'>)</span>,
    genome <span class='op'>=</span> <span class='st'>"hg38"</span>, mccs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>2</span>, <span class='fl'>4</span>, <span class='fl'>6</span>, <span class='fl'>8</span>, <span class='fl'>10</span><span class='op'>)</span>, mrgs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>10</span>, <span class='fl'>20</span>, <span class='fl'>30</span><span class='op'>)</span>,
    gtf <span class='op'>=</span> <span class='va'>gtf_path</span>, ucsc_chr <span class='op'>=</span> <span class='cn'>TRUE</span>, ignore.strand <span class='op'>=</span> <span class='cn'>TRUE</span>,
    exons_no_overlap <span class='op'>=</span> <span class='cn'>NULL</span>, bw_chr <span class='op'>=</span> <span class='st'>"chr"</span>
<span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "2021-08-24 17:25:20 - Obtaining mean coverage across 1 samples"
#&gt; [1] "2021-08-24 17:25:20 - chr21"
#&gt; [1] "2021-08-24 17:25:21 - chr22"
#&gt; [1] "2021-08-24 17:25:21 - Generating ERs for chr21"
#&gt; [1] "2021-08-24 17:25:24 - Generating ERs for chr22"
#&gt; [1] "2021-08-24 17:25:27 - Loading in GTF..."
#&gt; [1] "2021-08-24 17:26:03 - Obtaining non-overlapping exons"
#&gt; [1] "2021-08-24 17:26:05 - Calculating delta for ERs..."
#&gt; [1] "2021-08-24 17:26:06 - Obtaining optimal set of ERs..."</div><div class='input'>
<span class='va'>test_juncs</span> <span class='op'>&lt;-</span> <span class='fu'>SummarizedExperiment</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html'>rowRanges</a></span><span class='op'>(</span><span class='fu'>dasper</span><span class='fu'>::</span><span class='va'><a href='https://rdrr.io/pkg/dasper/man/junctions_example.html'>junctions_example</a></span><span class='op'>)</span>
<span class='va'>aers</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/warning.html'>suppressWarnings</a></span><span class='op'>(</span><span class='fu'><a href='annotatERs.html'>annotatERs</a></span><span class='op'>(</span><span class='va'>opt_ers</span><span class='op'>[[</span><span class='st'>"opt_ers"</span><span class='op'>]</span><span class='op'>]</span>,
    junc_data <span class='op'>=</span> <span class='va'>test_juncs</span>,
    gtf_path <span class='op'>=</span> <span class='va'>gtf_path</span>, ensembl <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "2021-08-24 17:26:06 - Obtaining co-ordinates of annotated exons and junctions from gtf/gff3..."
#&gt; [1] "2021-08-24 17:26:06 - Importing gtf/gff3 as a TxDb..."</div><div class='output co'>#&gt; <span class='message'>Import genomic features from the file as a GRanges object ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Prepare the 'metadata' data frame ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Make the TxDb object ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; [1] "2021-08-24 17:27:26 - Getting junction annotation using overlapping exons..."
#&gt; [1] "2021-08-24 17:27:28 - Tidying junction annotation..."
#&gt; [1] "2021-08-24 17:27:28 - Deriving junction categories..."
#&gt; [1] "2021-08-24 17:27:32 - done!"
#&gt; [1] "2021-08-24 17:27:32 - Finding junctions overlapping ers..."
#&gt; [1] "2021-08-24 17:27:58 - Generating a genomic state..."</div><div class='output co'>#&gt; <span class='message'>Import genomic features from the file as a GRanges object ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Prepare the 'metadata' data frame ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Make the TxDb object ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:1 mapping between keys and columns</span></div><div class='output co'>#&gt; [1] "2021-08-24 17:34:23 - Annotating the Expressed regions..."</div><div class='output co'>#&gt; <span class='message'>2021-08-24 17:34:24 annotateRegions: counting</span></div><div class='output co'>#&gt; <span class='message'>2021-08-24 17:34:24 annotateRegions: annotating</span></div><div class='output co'>#&gt; [1] "2021-08-24 17:35:06 - done!"</div><div class='input'><span class='va'>aers2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/warning.html'>suppressWarnings</a></span><span class='op'>(</span><span class='fu'><a href='annotatERs.html'>annotatERs</a></span><span class='op'>(</span><span class='va'>opt_ers2</span><span class='op'>[[</span><span class='st'>"opt_ers"</span><span class='op'>]</span><span class='op'>]</span>,
    junc_data <span class='op'>=</span> <span class='va'>test_juncs</span>,
    gtf_path <span class='op'>=</span> <span class='va'>gtf_path</span>, ensembl <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "2021-08-24 17:35:06 - Obtaining co-ordinates of annotated exons and junctions from gtf/gff3..."
#&gt; [1] "2021-08-24 17:35:06 - Importing gtf/gff3 as a TxDb..."</div><div class='output co'>#&gt; <span class='message'>Import genomic features from the file as a GRanges object ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Prepare the 'metadata' data frame ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Make the TxDb object ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; [1] "2021-08-24 17:36:22 - Getting junction annotation using overlapping exons..."
#&gt; [1] "2021-08-24 17:36:24 - Tidying junction annotation..."
#&gt; [1] "2021-08-24 17:36:24 - Deriving junction categories..."
#&gt; [1] "2021-08-24 17:36:28 - done!"
#&gt; [1] "2021-08-24 17:36:28 - Finding junctions overlapping ers..."
#&gt; [1] "2021-08-24 17:36:52 - Generating a genomic state..."</div><div class='output co'>#&gt; <span class='message'>Import genomic features from the file as a GRanges object ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Prepare the 'metadata' data frame ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Make the TxDb object ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:1 mapping between keys and columns</span></div><div class='output co'>#&gt; [1] "2021-08-24 17:43:08 - Annotating the Expressed regions..."</div><div class='output co'>#&gt; <span class='message'>2021-08-24 17:43:09 annotateRegions: counting</span></div><div class='output co'>#&gt; <span class='message'>2021-08-24 17:43:09 annotateRegions: annotating</span></div><div class='output co'>#&gt; [1] "2021-08-24 17:43:52 - done!"</div><div class='input'><span class='co'># }</span>
<span class='va'>example_bw_paths</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>bw_path</span>, <span class='va'>bw_path2</span><span class='op'>)</span>
<span class='va'>annot_ersl</span> <span class='op'>&lt;-</span> <span class='fu'>GenomicRanges</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html'>GRangesList</a></span><span class='op'>(</span><span class='va'>aers</span>, <span class='va'>aers2</span><span class='op'>)</span>

<span class='va'>example_cm</span> <span class='op'>&lt;-</span> <span class='fu'>get_count_matrix</span><span class='op'>(</span>bw_paths <span class='op'>=</span> <span class='va'>example_bw_paths</span>, annot_ers <span class='op'>=</span> <span class='va'>annot_ersl</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>The latest megadepth version is 1.1.0c</span></div><div class='output co'>#&gt; <span class='message'>megadepth has been installed to /github/home/bin</span></div><div class='output co'>#&gt; <span class='warning'>Warning: is.na() applied to non-(list or vector) of type 'S4'</span></div><div class='output co'>#&gt; <span class='warning'>Warning: is.na() applied to non-(list or vector) of type 'S4'</span></div><div class='input'><span class='va'>example_cm</span>
</div><div class='output co'>#&gt; class: RangedSummarizedExperiment 
#&gt; dim: 5716 2 
#&gt; metadata(0):
#&gt; assays(1): ''
#&gt; rownames: NULL
#&gt; rowData names(0):
#&gt; colnames(2): BFC5 BFC6
#&gt; colData names(1): bw_paths</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Emmanuel Olagbaju.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


