<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>ODER: Optimising the Definition of Expressed Regions — ODER • ODER</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="ODER: Optimising the Definition of Expressed Regions — ODER"><meta property="og:description" content="The aim of ODER is to identify previously unannotated expressed regions (ERs)
using RNA-sequencing data. For this purpose, ODER defines and optimises the
definition of ERs, then connected these ERs to genes using junction data. In
this way, ODER improves gene annotation. Gene annotation is a staple input of
many bioinformatic pipelines and a more complete gene annotation can enable
more accurate interpretation of disease associated variants.
Returns the optimum definition of the expressed regions by finding the ideal
MCC (Mean Coverage Cutoff) and MRG (Max Region Gap). The combination of MCC
and MRG that returns the expressed region with the smallest exon delta is the
most ideal."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ODER</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.29</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/ODER_overview.html">ODER: Optimising the Definition of Expressed Regions</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/eolagbaju/ODER/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>ODER: Optimising the Definition of Expressed Regions</h1>
    <small class="dont-index">Source: <a href="https://github.com/eolagbaju/ODER/blob/master/R/ODER-package.R"><code>R/ODER-package.R</code></a>, <a href="https://github.com/eolagbaju/ODER/blob/master/R/ODER.R"><code>R/ODER.R</code></a></small>
    <div class="hidden name"><code>ODER.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The aim of ODER is to identify previously unannotated expressed regions (ERs)
using RNA-sequencing data. For this purpose, ODER defines and optimises the
definition of ERs, then connected these ERs to genes using junction data. In
this way, ODER improves gene annotation. Gene annotation is a staple input of
many bioinformatic pipelines and a more complete gene annotation can enable
more accurate interpretation of disease associated variants.</p>
<p>Returns the optimum definition of the expressed regions by finding the ideal
MCC (Mean Coverage Cutoff) and MRG (Max Region Gap). The combination of MCC
and MRG that returns the expressed region with the smallest exon delta is the
most ideal.</p>
    </div>

    <div class="ref-usage sourceCode"><pre class="sourceCode r"><code><span class="fu">ODER</span><span class="op">(</span>
  <span class="va">bw_paths</span>,
  <span class="va">auc_raw</span>,
  <span class="va">auc_target</span>,
  chrs <span class="op">=</span> <span class="st">""</span>,
  genome <span class="op">=</span> <span class="st">"hg38"</span>,
  <span class="va">mccs</span>,
  <span class="va">mrgs</span>,
  gtf <span class="op">=</span> <span class="cn">NULL</span>,
  <span class="va">ucsc_chr</span>,
  <span class="va">ignore.strand</span>,
  exons_no_overlap <span class="op">=</span> <span class="cn">NULL</span>,
  biotype <span class="op">=</span> <span class="st">"Non-overlapping"</span>,
  bw_chr <span class="op">=</span> <span class="st">"chr"</span>,
  file_type <span class="op">=</span> <span class="st">"non-stranded"</span>,
  bw_pos <span class="op">=</span> <span class="cn">NULL</span>,
  bw_neg <span class="op">=</span> <span class="cn">NULL</span>,
  auc_raw_pos <span class="op">=</span> <span class="cn">NULL</span>,
  auc_raw_neg <span class="op">=</span> <span class="cn">NULL</span>
<span class="op">)</span></code></pre></div>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments"><colgroup><col class="name"><col class="desc"></colgroup><tr><th>bw_paths</th>
      <td><p>path(s) to bigwig file(s) with the RNA-seq data that you want
the #'   coverage of.</p></td>
    </tr><tr><th>auc_raw</th>
      <td><p>vector containing AUCs(Area Under Coverage) matching the order
of bigwig path(s).</p></td>
    </tr><tr><th>auc_target</th>
      <td><p>total AUC to normalise all samples to e.g. 40e6 * 100 would
be the estimated total auc for sample sequenced to 40 million reads of
100bp in length.</p></td>
    </tr><tr><th>chrs</th>
      <td><p>chromosomes to obtain mean coverage for, default is "" giving
every chromosome. Can take UCSC format(chrs = "chr1") or just the
chromosome i.e. chrs = c(1,X)</p></td>
    </tr><tr><th>genome</th>
      <td><p>the UCSC genome you want to use, the default is hg38.</p></td>
    </tr><tr><th>mccs</th>
      <td><p>mean coverage cut-offs to apply.</p></td>
    </tr><tr><th>mrgs</th>
      <td><p>max region gaps to apply.</p></td>
    </tr><tr><th>gtf</th>
      <td><p>Either a string containg the path to a .gtf file or a pre-imported
gtf using <code>rtracklayer::import</code> .</p></td>
    </tr><tr><th>ucsc_chr</th>
      <td><p>logical scalar, determining whether to add "chr" prefix to
the seqnames of non-overlapping exons and change "chrMT" -&gt; "chrM". Note,
if set to TRUE and seqnames already have "chr", it will not add another.</p></td>
    </tr><tr><th>ignore.strand</th>
      <td><p>logical value for input into
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html">findOverlaps</a></code>, default is True.</p></td>
    </tr><tr><th>exons_no_overlap</th>
      <td><p>Optimum set of exons to help calculate deltas</p></td>
    </tr><tr><th>biotype</th>
      <td><p>Filters the GTF file passed in to what would be considered the
"Gold Standard" exons. The Default is "Non-overlapping" but the options are:
"Non-overlapping" (exons that don't intersect each other),
"Three Prime" (3' UTR), "Five Prime" (5' UTR), "Internal" (Internal coding),
"lncRNA" (Long Non-Coding RNA), "ncRNA" (Non-Coding RNA) and "Pseudogene"</p></td>
    </tr><tr><th>bw_chr</th>
      <td><p>specifies whether the bigwig files has the chromosomes labelled
with a "chr" preceding the chromosome i.e. "chr1" vs "1". Can be either
"chr" or "nochr" with "chr" being the default.</p></td>
    </tr><tr><th>file_type</th>
      <td><p>Describes if the BigWigs are stranded or not. Either
"stranded" or non-stranded</p></td>
    </tr><tr><th>bw_pos</th>
      <td><p>positive strand bigwig file</p></td>
    </tr><tr><th>bw_neg</th>
      <td><p>negative strand bigwig file</p></td>
    </tr><tr><th>auc_raw_pos</th>
      <td><p>vector containing AUCs(Area Under Coverage) matching the
order of the positive bigwig paths.</p></td>
    </tr><tr><th>auc_raw_neg</th>
      <td><p>vector containing AUCs(Area Under Coverage) matching the
order of the negative bigwig paths.</p></td>
    </tr></table><h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>list containing optimised ERs, optimal pair of MCC/MRGs and
<code>delta_df</code></p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <div class="ref-examples sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">rec_url</span> <span class="op">&lt;-</span> <span class="fu">recount</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study</a></span><span class="op">(</span></span>
<span class="r-in">    project <span class="op">=</span> <span class="st">"SRP012682"</span>,</span>
<span class="r-in">    type <span class="op">=</span> <span class="st">"samples"</span>,</span>
<span class="r-in">    download <span class="op">=</span> <span class="cn">FALSE</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Setting options('download.file.method.GEOquery'='auto')</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Setting options('GEOquery.inmemory.gpl'=FALSE)</span>
<span class="r-in"><span class="co"># file_cache is an internal function to download a bigwig file from a link</span></span>
<span class="r-in"><span class="co"># if the file has been downloaded recently, it will be retrieved from a cache</span></span>
<span class="r-in"><span class="va">bw_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="file_cache.html">file_cache</a></span><span class="op">(</span><span class="va">rec_url</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/exists.html">exists</a></span><span class="op">(</span><span class="st">"gtf_path"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in">    <span class="va">gtf_url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span></span>
<span class="r-in">        <span class="st">"http://ftp.ensembl.org/pub/release-103/gtf/"</span>,</span>
<span class="r-in">        <span class="st">"homo_sapiens/Homo_sapiens.GRCh38.103.chr.gtf.gz"</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-in">    <span class="va">gtf_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="file_cache.html">file_cache</a></span><span class="op">(</span><span class="va">gtf_url</span><span class="op">)</span></span>
<span class="r-in"><span class="op">}</span></span>
<span class="r-in"><span class="co"># As of rtracklayer 1.25.16, BigWig is not supported on Windows.</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">gtex_SRP012682_SRX222703_lung_auc_1</span>, package <span class="op">=</span> <span class="st">"ODER"</span><span class="op">)</span></span>
<span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu">xfun</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/xfun/man/os.html">is_windows</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in">    <span class="va">opt_ers</span> <span class="op">&lt;-</span> <span class="fu">ODER</span><span class="op">(</span></span>
<span class="r-in">        bw_paths <span class="op">=</span> <span class="va">bw_path</span>,</span>
<span class="r-in">        auc_raw <span class="op">=</span> <span class="va">gtex_SRP012682_SRX222703_lung_auc_1</span>,</span>
<span class="r-in">        auc_target <span class="op">=</span> <span class="fl">40e6</span> <span class="op">*</span> <span class="fl">100</span>, chrs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chr21"</span>, <span class="st">"chr22"</span><span class="op">)</span>,</span>
<span class="r-in">        genome <span class="op">=</span> <span class="st">"hg38"</span>, mccs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">10</span><span class="op">)</span>, mrgs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">20</span><span class="op">)</span>,</span>
<span class="r-in">        gtf <span class="op">=</span> <span class="va">gtf_path</span>, ucsc_chr <span class="op">=</span> <span class="cn">TRUE</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span class="r-in">        exons_no_overlap <span class="op">=</span> <span class="cn">NULL</span>, bw_chr <span class="op">=</span> <span class="st">"chr"</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in">    <span class="va">opt_ers</span></span>
<span class="r-in"><span class="op">}</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BiocGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘BiocGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:stats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     IQR, mad, sd, var, xtabs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Filter, Find, Map, Position, Reduce, anyDuplicated, append,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.data.frame, basename, cbind, colnames, dirname, do.call,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     tapply, union, unique, unsplit, which.max, which.min</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: S4Vectors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: stats4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘S4Vectors’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     I, expand.grid, unname</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2021-10-07 15:46:18 - Obtaining mean coverage across 1 samples</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2021-10-07 15:46:18 - chr21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2021-10-07 15:46:19 - chr22</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2021-10-07 15:46:20 - Generating ERs for chr21</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2021-10-07 15:46:23 - Generating ERs for chr22</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2021-10-07 15:46:25 - Loading in GTF...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2021-10-07 15:47:05 - Obtaining non-overlapping exons</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2021-10-07 15:47:07 - Calculating delta for ERs...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2021-10-07 15:47:07 - Obtaining optimal set of ERs...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $opt_ers</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 15977 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           seqnames            ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [1]    chr21   5032176-5032217      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [2]    chr21   5033408-5033425      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [3]    chr21   5034717-5034756      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [4]    chr21   5035188-5035189      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [5]    chr21   5036577-5036581      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       ...      ...               ...    ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [15973]    chr22 50798996-50799149      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [15974]    chr22 50799209-50799284      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [15975]    chr22 50799669-50799688      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [15976]    chr22 50799717-50799744      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [15977]    chr22 50800460-50800587      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $opt_mcc_mrg</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "mcc_10" "mrg_20"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $deltas</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 4 × 7</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mcc   mrg     sum  mean median n_eq_0 propor_eq_0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span>     5    10 2<span style="text-decoration: underline;">187</span>300  964.   150.    322       0.142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span>     5    20 1<span style="text-decoration: underline;">892</span>386  898.   141     335       0.159</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span>    10    10 1<span style="text-decoration: underline;">771</span>837 <span style="text-decoration: underline;">1</span>009.   139     316       0.180</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span>    10    20 1<span style="text-decoration: underline;">457</span>726  911.   118     323       0.202</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Emmanuel Olagbaju, David Zhang.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.9000.9000.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

