<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Generates the optimum expressed regions — ODER • ODER</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Generates the optimum expressed regions — ODER" />
<meta property="og:description" content="Returns the optimum definition of the expressed regions by finding the ideal
MCC (Mean Coverage Cutoff) and MRG (Max Region Gap)" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ODER</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ODERflow.html">ODER workflow</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/eolagbaju/ODER/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generates the optimum expressed regions</h1>
    <small class="dont-index">Source: <a href='https://github.com/eolagbaju/ODER/blob/master/R/ODER.R'><code>R/ODER.R</code></a></small>
    <div class="hidden name"><code>ODER.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Returns the optimum definition of the expressed regions by finding the ideal
MCC (Mean Coverage Cutoff) and MRG (Max Region Gap)</p>
    </div>

    <pre class="usage"><span class='fu'>ODER</span><span class='op'>(</span>
  <span class='va'>bw_paths</span>,
  <span class='va'>auc_raw</span>,
  <span class='va'>auc_target</span>,
  chrs <span class='op'>=</span> <span class='st'>""</span>,
  genome <span class='op'>=</span> <span class='st'>"hg38"</span>,
  <span class='va'>mccs</span>,
  <span class='va'>mrgs</span>,
  gtf <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>ucsc_chr</span>,
  <span class='va'>ignore.strand</span>,
  exons_no_overlap <span class='op'>=</span> <span class='cn'>NULL</span>,
  biotype <span class='op'>=</span> <span class='st'>"Non-overlapping"</span>,
  bw_chr <span class='op'>=</span> <span class='st'>"chr"</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>bw_paths</th>
      <td><p>paths to bigwig files with the RNA-seq data that you want the
coverage of.</p></td>
    </tr>
    <tr>
      <th>auc_raw</th>
      <td><p>vector containing AUCs(Area Under Coverage) matching the order
of bigwig paths.</p></td>
    </tr>
    <tr>
      <th>auc_target</th>
      <td><p>total AUC to normalise all samples to e.g. 40e6 * 100
would be the estimated total auc for sample sequenced to 40 million reads
of 100bp in length.</p></td>
    </tr>
    <tr>
      <th>chrs</th>
      <td><p>chromosomes to obtain mean coverage for, default is "" giving
every chromosome. Can take UCSC format(chrs = "chr1")
or just the chromosome i.e. chrs = c(1,X)</p></td>
    </tr>
    <tr>
      <th>genome</th>
      <td><p>the UCSC genome you want to use, the default is hg38.</p></td>
    </tr>
    <tr>
      <th>mccs</th>
      <td><p>mean coverage cut-offs to apply.</p></td>
    </tr>
    <tr>
      <th>mrgs</th>
      <td><p>max region gaps to apply.</p></td>
    </tr>
    <tr>
      <th>gtf</th>
      <td><p>Either a string containg the path to a .gtf file or a pre-imported
gtf using <code><a href='https://rdrr.io/pkg/rtracklayer/man/export.html'>import</a></code>.</p></td>
    </tr>
    <tr>
      <th>ucsc_chr</th>
      <td><p>logical scalar, determining whether to add "chr" prefix to
the seqnames of non-overlapping exons and change "chrMT" -&gt; "chrM". Note,
if set to TRUE and seqnames already have "chr", it will not add another.</p></td>
    </tr>
    <tr>
      <th>ignore.strand</th>
      <td><p>logical value for input into
<code><a href='https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html'>findOverlaps</a></code>, default is True.</p></td>
    </tr>
    <tr>
      <th>exons_no_overlap</th>
      <td><p>Optimum set of exons to help calculate deltas</p></td>
    </tr>
    <tr>
      <th>biotype</th>
      <td><p>Filters the GTF file passed in to what would be considered the
"Gold Standard" exons. The Default is "Non-overlapping" but the options are:
"Non-overlapping" (exons that don't intersect each other),
"Three Prime" (3' UTR), "Five Prime" (5' UTR), "Internal" (Internal coding),
"lncRNA" (Long Non-Coding RNA), "ncRNA" (Non-Coding RNA) and "Pseudogene"</p></td>
    </tr>
    <tr>
      <th>bw_chr</th>
      <td><p>specifies whether the bigwig files has the chromosomes labelled
with a "chr" preceding the chromosome i.e. "chr1" vs "1". Can be either
"chr" or "nochr" with "chr" being the default.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>list containing optimised ERs, optimal pair of MCC/MRGs and
<code>delta_df</code></p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># \dontshow{</span>
<span class='va'>url</span> <span class='op'>&lt;-</span> <span class='fu'>recount</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/recount/man/download_study.html'>download_study</a></span><span class='op'>(</span>
    project <span class='op'>=</span> <span class='st'>"SRP012682"</span>,
    type <span class='op'>=</span> <span class='st'>"samples"</span>,
    download <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span> <span class='co'># .file_cache is an internal function to download a bigwig file from a link</span>
</div><div class='output co'>#&gt; <span class='message'>Setting options('download.file.method.GEOquery'='auto')</span></div><div class='output co'>#&gt; <span class='message'>Setting options('GEOquery.inmemory.gpl'=FALSE)</span></div><div class='input'><span class='co'># if the file has been downloaded recently, it will be retrieved from a cache</span>

<span class='va'>bw_path</span> <span class='op'>&lt;-</span> <span class='fu'>ODER</span><span class='fu'>:::</span><span class='fu'>.file_cache</span><span class='op'>(</span><span class='va'>url</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>)</span>
<span class='va'>gtf_url</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span>
    <span class='st'>"http://ftp.ensembl.org/pub/release-103/gtf/"</span>,
    <span class='st'>"homo_sapiens/Homo_sapiens.GRCh38.103.chr.gtf.gz"</span>
<span class='op'>)</span>
<span class='va'>gtf_path</span> <span class='op'>&lt;-</span> <span class='fu'>ODER</span><span class='fu'>:::</span><span class='fu'>.file_cache</span><span class='op'>(</span><span class='va'>gtf_url</span><span class='op'>)</span>
<span class='co'># }</span>

<span class='va'>opt_ers</span> <span class='op'>&lt;-</span> <span class='fu'>ODER</span><span class='op'>(</span>
    bw_paths <span class='op'>=</span> <span class='va'>bw_path</span>, auc_raw <span class='op'>=</span> <span class='va'>auc_example</span>,
    auc_target <span class='op'>=</span> <span class='fl'>40e6</span> <span class='op'>*</span> <span class='fl'>100</span>, chrs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"chr21"</span>, <span class='st'>"chr22"</span><span class='op'>)</span>,
    genome <span class='op'>=</span> <span class='st'>"hg38"</span>, mccs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>5</span>, <span class='fl'>10</span><span class='op'>)</span>, mrgs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>10</span>, <span class='fl'>20</span><span class='op'>)</span>,
    gtf <span class='op'>=</span> <span class='va'>gtf_path</span>, ucsc_chr <span class='op'>=</span> <span class='cn'>TRUE</span>, ignore.strand <span class='op'>=</span> <span class='cn'>TRUE</span>,
    exons_no_overlap <span class='op'>=</span> <span class='cn'>NULL</span>, bw_chr <span class='op'>=</span> <span class='st'>"chr"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocGenerics</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: parallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘BiocGenerics’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:parallel’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
#&gt; <span class='message'>    clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
#&gt; <span class='message'>    parLapplyLB, parRapply, parSapply, parSapplyLB</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:stats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    IQR, mad, sd, var, xtabs</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
#&gt; <span class='message'>    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
#&gt; <span class='message'>    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
#&gt; <span class='message'>    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
#&gt; <span class='message'>    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
#&gt; <span class='message'>    union, unique, unsplit, which.max, which.min</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: S4Vectors</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: stats4</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘S4Vectors’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; [1] "2021-08-24 21:47:54 - Obtaining mean coverage across 1 samples"
#&gt; [1] "2021-08-24 21:47:54 - chr21"
#&gt; [1] "2021-08-24 21:47:55 - chr22"
#&gt; [1] "2021-08-24 21:47:56 - Generating ERs for chr21"
#&gt; [1] "2021-08-24 21:47:59 - Generating ERs for chr22"
#&gt; [1] "2021-08-24 21:48:01 - Loading in GTF..."
#&gt; [1] "2021-08-24 21:48:43 - Obtaining non-overlapping exons"
#&gt; [1] "2021-08-24 21:48:45 - Calculating delta for ERs..."
#&gt; [1] "2021-08-24 21:48:46 - Obtaining optimal set of ERs..."</div><div class='input'>
<span class='va'>opt_ers</span>
</div><div class='output co'>#&gt; $opt_ers
#&gt; GRanges object with 15977 ranges and 0 metadata columns:
#&gt;           seqnames            ranges strand
#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;       [1]    chr21   5032176-5032217      *
#&gt;       [2]    chr21   5033408-5033425      *
#&gt;       [3]    chr21   5034717-5034756      *
#&gt;       [4]    chr21   5035188-5035189      *
#&gt;       [5]    chr21   5036577-5036581      *
#&gt;       ...      ...               ...    ...
#&gt;   [15973]    chr22 50798996-50799149      *
#&gt;   [15974]    chr22 50799209-50799284      *
#&gt;   [15975]    chr22 50799669-50799688      *
#&gt;   [15976]    chr22 50799717-50799744      *
#&gt;   [15977]    chr22 50800460-50800587      *
#&gt;   -------
#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths
#&gt; 
#&gt; $opt_mcc_mrg
#&gt; [1] "mcc_10" "mrg_20"
#&gt; 
#&gt; $deltas
#&gt; <span style='color: #555555;'># A tibble: 4 x 7</span><span>
#&gt;     mcc   mrg     sum  mean median n_eq_0 propor_eq_0
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>   </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>       </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #555555;'>1</span><span>     5    10 2</span><span style='text-decoration: underline;'>187</span><span>300  964.   150.    322       0.142
#&gt; </span><span style='color: #555555;'>2</span><span>     5    20 1</span><span style='text-decoration: underline;'>892</span><span>386  898.   141     335       0.159
#&gt; </span><span style='color: #555555;'>3</span><span>    10    10 1</span><span style='text-decoration: underline;'>771</span><span>837 </span><span style='text-decoration: underline;'>1</span><span>009.   139     316       0.180
#&gt; </span><span style='color: #555555;'>4</span><span>    10    20 1</span><span style='text-decoration: underline;'>457</span><span>726  911.   118     323       0.202
#&gt; </div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Emmanuel Olagbaju.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


