<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Generates the optimum expressed regions — ODER_strand • ODER</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Generates the optimum expressed regions — ODER_strand" />
<meta property="og:description" content="Generates the optimum expressed regions" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ODER</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/ODERflow.html">ODER workflow</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/eolagbaju/ODER/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generates the optimum expressed regions</h1>
    <small class="dont-index">Source: <a href='https://github.com/eolagbaju/ODER/blob/master/R/ODER.R'><code>R/ODER.R</code></a></small>
    <div class="hidden name"><code>ODER_strand.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Generates the optimum expressed regions</p>
    </div>

    <pre class="usage"><span class='fu'>ODER_strand</span><span class='op'>(</span>
  <span class='va'>bw_pos</span>,
  <span class='va'>bw_neg</span>,
  <span class='va'>auc_raw_pos</span>,
  <span class='va'>auc_raw_neg</span>,
  <span class='va'>auc_target</span>,
  chrs <span class='op'>=</span> <span class='st'>""</span>,
  genome <span class='op'>=</span> <span class='st'>"hg38"</span>,
  <span class='va'>mccs</span>,
  <span class='va'>mrgs</span>,
  gtf <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>ucsc_chr</span>,
  <span class='va'>ignore.strand</span>,
  exons_no_overlap <span class='op'>=</span> <span class='cn'>NULL</span>,
  bw_chr <span class='op'>=</span> <span class='st'>"chr"</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>bw_pos</th>
      <td><p>positive strand bigwig file</p></td>
    </tr>
    <tr>
      <th>bw_neg</th>
      <td><p>negative strand bigwig file</p></td>
    </tr>
    <tr>
      <th>auc_raw_pos</th>
      <td><p>vector containing AUCs(Area Under Coverage) matching the order
of bigwig paths.</p></td>
    </tr>
    <tr>
      <th>auc_raw_neg</th>
      <td><p>vector containing AUCs(Area Under Coverage) matching the order
of bigwig paths.</p></td>
    </tr>
    <tr>
      <th>auc_target</th>
      <td><p>total AUC to normalise all samples to e.g. 40e6 * 100
would be the estimated total auc for sample sequenced to 40 million reads
of 100bp in length.</p></td>
    </tr>
    <tr>
      <th>chrs</th>
      <td><p>chromosomes to obtain mean coverage for, default is "" giving
every chromosome. Can take UCSC format(chrs = "chr1")
or just the chromosome i.e. chrs = c(1,X)</p></td>
    </tr>
    <tr>
      <th>genome</th>
      <td><p>the UCSC genome you want to use, the default is hg38.</p></td>
    </tr>
    <tr>
      <th>mccs</th>
      <td><p>mean coverage cut-offs to apply.</p></td>
    </tr>
    <tr>
      <th>mrgs</th>
      <td><p>max region gaps to apply.</p></td>
    </tr>
    <tr>
      <th>gtf</th>
      <td><p>Either a string containg the path to a .gtf file or a pre-imported
gtf using <code><a href='https://rdrr.io/pkg/rtracklayer/man/export.html'>import</a></code>.</p></td>
    </tr>
    <tr>
      <th>ucsc_chr</th>
      <td><p>logical scalar, determining whether to add "chr" prefix to
the seqnames of non-overlapping exons and change "chrMT" -&gt; "chrM". Note,
if set to TRUE and seqnames already have "chr", it will not add another.</p></td>
    </tr>
    <tr>
      <th>ignore.strand</th>
      <td><p>logical value for input into
<code><a href='https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html'>findOverlaps</a></code>, default is True.</p></td>
    </tr>
    <tr>
      <th>exons_no_overlap</th>
      <td><p>Optimum set of exons to help calculate deltas</p></td>
    </tr>
    <tr>
      <th>bw_chr</th>
      <td><p>specifies whether the bigwig files has the chromosomes labelled
with a "chr" preceding the chromosome i.e. "chr1" vs "1". Can be either
"chr" or "nochr" with "chr" being the default.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>list containing optimised ERs, optimal pair of MCC/MRGs and
<code>delta_df</code></p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'><a href='https://magrittr.tidyverse.org'>"magrittr"</a></span><span class='op'>)</span>
<span class='va'>gtex_metadata</span> <span class='op'>&lt;-</span> <span class='fu'>recount</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/recount/man/all_metadata.html'>all_metadata</a></span><span class='op'>(</span><span class='st'>"gtex"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>2021-08-24 21:48:46 downloading the metadata to /tmp/RtmpfgXOQD/metadata_clean_gtex.Rdata</span></div><div class='input'><span class='va'>gtex_metadata</span> <span class='op'>&lt;-</span> <span class='va'>gtex_metadata</span> <span class='op'>%&gt;%</span>
    <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
    <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>project</span> <span class='op'>==</span> <span class='st'>"SRP012682"</span><span class='op'>)</span>

<span class='va'>gtf_url</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span>
    <span class='st'>"http://ftp.ensembl.org/pub/release-103/gtf/"</span>,
    <span class='st'>"homo_sapiens/Homo_sapiens.GRCh38.103.chr.gtf.gz"</span>
<span class='op'>)</span>
<span class='va'>gtf_path</span> <span class='op'>&lt;-</span> <span class='fu'>ODER</span><span class='fu'>:::</span><span class='fu'>.file_cache</span><span class='op'>(</span><span class='va'>gtf_url</span><span class='op'>)</span>

<span class='va'>url</span> <span class='op'>&lt;-</span> <span class='fu'>recount</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/recount/man/download_study.html'>download_study</a></span><span class='op'>(</span>
    project <span class='op'>=</span> <span class='st'>"SRP012682"</span>,
    type <span class='op'>=</span> <span class='st'>"samples"</span>,
    download <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span>
<span class='va'>bw_plus</span> <span class='op'>&lt;-</span> <span class='fu'>ODER</span><span class='fu'>:::</span><span class='fu'>.file_cache</span><span class='op'>(</span><span class='va'>url</span><span class='op'>[</span><span class='fl'>58</span><span class='op'>]</span><span class='op'>)</span>
<span class='va'>bw_minus</span> <span class='op'>&lt;-</span> <span class='fu'>ODER</span><span class='fu'>:::</span><span class='fu'>.file_cache</span><span class='op'>(</span><span class='va'>url</span><span class='op'>[</span><span class='fl'>84</span><span class='op'>]</span><span class='op'>)</span>

<span class='va'>opt_ers</span> <span class='op'>&lt;-</span> <span class='fu'>ODER_strand</span><span class='op'>(</span>
    bw_pos <span class='op'>=</span> <span class='va'>bw_plus</span>, bw_neg <span class='op'>=</span> <span class='va'>bw_minus</span>,
    auc_raw_pos <span class='op'>=</span> <span class='va'>gtex_metadata</span><span class='op'>[[</span><span class='st'>"auc"</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span><span class='fl'>58</span><span class='op'>]</span>, auc_raw_neg <span class='op'>=</span> <span class='va'>gtex_metadata</span><span class='op'>[[</span><span class='st'>"auc"</span><span class='op'>]</span><span class='op'>]</span><span class='op'>[</span><span class='fl'>84</span><span class='op'>]</span>,
    auc_target <span class='op'>=</span> <span class='fl'>40e6</span> <span class='op'>*</span> <span class='fl'>100</span>, chrs <span class='op'>=</span> <span class='st'>""</span>, genome <span class='op'>=</span> <span class='st'>"hg38"</span>,
    mccs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>5</span>, <span class='fl'>10</span><span class='op'>)</span>, mrgs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>10</span>, <span class='fl'>20</span><span class='op'>)</span>, gtf <span class='op'>=</span> <span class='va'>gtf_path</span>, ucsc_chr <span class='op'>=</span> <span class='cn'>TRUE</span>,
    ignore.strand <span class='op'>=</span> <span class='cn'>FALSE</span>, exons_no_overlap <span class='op'>=</span> <span class='cn'>NULL</span>, bw_chr <span class='op'>=</span> <span class='st'>"chr"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "2021-08-24 21:48:48 - Obtaining mean coverage across 1 samples"
#&gt; [1] "2021-08-24 21:48:48 - chr1"
#&gt; [1] "2021-08-24 21:48:51 - chr2"
#&gt; [1] "2021-08-24 21:48:53 - chr3"
#&gt; [1] "2021-08-24 21:48:54 - chr4"
#&gt; [1] "2021-08-24 21:48:55 - chr5"
#&gt; [1] "2021-08-24 21:48:56 - chr6"
#&gt; [1] "2021-08-24 21:48:57 - chr7"
#&gt; [1] "2021-08-24 21:48:58 - chr8"
#&gt; [1] "2021-08-24 21:48:59 - chr9"
#&gt; [1] "2021-08-24 21:49:00 - chr10"
#&gt; [1] "2021-08-24 21:49:01 - chr11"
#&gt; [1] "2021-08-24 21:49:02 - chr12"
#&gt; [1] "2021-08-24 21:49:03 - chr13"
#&gt; [1] "2021-08-24 21:49:03 - chr14"
#&gt; [1] "2021-08-24 21:49:04 - chr15"
#&gt; [1] "2021-08-24 21:49:04 - chr16"
#&gt; [1] "2021-08-24 21:49:05 - chr17"
#&gt; [1] "2021-08-24 21:49:06 - chr18"
#&gt; [1] "2021-08-24 21:49:07 - chr19"
#&gt; [1] "2021-08-24 21:49:08 - chr20"
#&gt; [1] "2021-08-24 21:49:08 - chr21"
#&gt; [1] "2021-08-24 21:49:09 - chr22"
#&gt; [1] "2021-08-24 21:49:09 - chrX"
#&gt; [1] "2021-08-24 21:49:10 - chrY"
#&gt; [1] "2021-08-24 21:49:10 - chrM"
#&gt; [1] "2021-08-24 21:49:10 - Obtaining mean coverage across 1 samples"
#&gt; [1] "2021-08-24 21:49:10 - chr1"
#&gt; [1] "2021-08-24 21:49:12 - chr2"
#&gt; [1] "2021-08-24 21:49:13 - chr3"
#&gt; [1] "2021-08-24 21:49:16 - chr4"
#&gt; [1] "2021-08-24 21:49:16 - chr5"
#&gt; [1] "2021-08-24 21:49:17 - chr6"
#&gt; [1] "2021-08-24 21:49:18 - chr7"
#&gt; [1] "2021-08-24 21:49:19 - chr8"
#&gt; [1] "2021-08-24 21:49:20 - chr9"
#&gt; [1] "2021-08-24 21:49:21 - chr10"
#&gt; [1] "2021-08-24 21:49:22 - chr11"
#&gt; [1] "2021-08-24 21:49:23 - chr12"
#&gt; [1] "2021-08-24 21:49:25 - chr13"
#&gt; [1] "2021-08-24 21:49:25 - chr14"
#&gt; [1] "2021-08-24 21:49:26 - chr15"
#&gt; [1] "2021-08-24 21:49:27 - chr16"
#&gt; [1] "2021-08-24 21:49:27 - chr17"
#&gt; [1] "2021-08-24 21:49:29 - chr18"
#&gt; [1] "2021-08-24 21:49:30 - chr19"
#&gt; [1] "2021-08-24 21:49:31 - chr20"
#&gt; [1] "2021-08-24 21:49:31 - chr21"
#&gt; [1] "2021-08-24 21:49:32 - chr22"
#&gt; [1] "2021-08-24 21:49:32 - chrX"
#&gt; [1] "2021-08-24 21:49:33 - chrY"
#&gt; [1] "2021-08-24 21:49:33 - chrM"
#&gt; [1] "2021-08-24 21:49:33 - Generating ERs for chr1"
#&gt; [1] "2021-08-24 21:49:38 - Generating ERs for chr2"
#&gt; [1] "2021-08-24 21:49:43 - Generating ERs for chr3"
#&gt; [1] "2021-08-24 21:49:45 - Generating ERs for chr4"
#&gt; [1] "2021-08-24 21:49:47 - Generating ERs for chr5"
#&gt; [1] "2021-08-24 21:49:49 - Generating ERs for chr6"
#&gt; [1] "2021-08-24 21:49:51 - Generating ERs for chr7"
#&gt; [1] "2021-08-24 21:49:52 - Generating ERs for chr8"
#&gt; [1] "2021-08-24 21:49:55 - Generating ERs for chr9"
#&gt; [1] "2021-08-24 21:49:56 - Generating ERs for chr10"
#&gt; [1] "2021-08-24 21:49:58 - Generating ERs for chr11"
#&gt; [1] "2021-08-24 21:49:59 - Generating ERs for chr12"
#&gt; [1] "2021-08-24 21:50:01 - Generating ERs for chr13"
#&gt; [1] "2021-08-24 21:50:02 - Generating ERs for chr14"
#&gt; [1] "2021-08-24 21:50:03 - Generating ERs for chr15"
#&gt; [1] "2021-08-24 21:50:05 - Generating ERs for chr16"
#&gt; [1] "2021-08-24 21:50:06 - Generating ERs for chr17"
#&gt; [1] "2021-08-24 21:50:08 - Generating ERs for chr18"
#&gt; [1] "2021-08-24 21:50:09 - Generating ERs for chr19"
#&gt; [1] "2021-08-24 21:50:11 - Generating ERs for chr20"
#&gt; [1] "2021-08-24 21:50:12 - Generating ERs for chr21"
#&gt; [1] "2021-08-24 21:50:12 - Generating ERs for chr22"
#&gt; [1] "2021-08-24 21:50:13 - Generating ERs for chrX"
#&gt; [1] "2021-08-24 21:50:14 - Generating ERs for chrY"
#&gt; [1] "2021-08-24 21:50:15 - Generating ERs for chrM"
#&gt; [1] "2021-08-24 21:50:16 - Generating ERs for chr1"
#&gt; [1] "2021-08-24 21:50:19 - Generating ERs for chr2"
#&gt; [1] "2021-08-24 21:50:22 - Generating ERs for chr3"
#&gt; [1] "2021-08-24 21:50:24 - Generating ERs for chr4"
#&gt; [1] "2021-08-24 21:50:27 - Generating ERs for chr5"
#&gt; [1] "2021-08-24 21:50:29 - Generating ERs for chr6"
#&gt; [1] "2021-08-24 21:50:30 - Generating ERs for chr7"
#&gt; [1] "2021-08-24 21:50:33 - Generating ERs for chr8"
#&gt; [1] "2021-08-24 21:50:34 - Generating ERs for chr9"
#&gt; [1] "2021-08-24 21:50:35 - Generating ERs for chr10"
#&gt; [1] "2021-08-24 21:50:37 - Generating ERs for chr11"
#&gt; [1] "2021-08-24 21:50:38 - Generating ERs for chr12"
#&gt; [1] "2021-08-24 21:50:41 - Generating ERs for chr13"
#&gt; [1] "2021-08-24 21:50:42 - Generating ERs for chr14"
#&gt; [1] "2021-08-24 21:50:43 - Generating ERs for chr15"
#&gt; [1] "2021-08-24 21:50:44 - Generating ERs for chr16"
#&gt; [1] "2021-08-24 21:50:45 - Generating ERs for chr17"
#&gt; [1] "2021-08-24 21:50:47 - Generating ERs for chr18"
#&gt; [1] "2021-08-24 21:50:47 - Generating ERs for chr19"
#&gt; [1] "2021-08-24 21:50:49 - Generating ERs for chr20"
#&gt; [1] "2021-08-24 21:50:49 - Generating ERs for chr21"
#&gt; [1] "2021-08-24 21:50:50 - Generating ERs for chr22"
#&gt; [1] "2021-08-24 21:50:51 - Generating ERs for chrX"
#&gt; [1] "2021-08-24 21:50:52 - Generating ERs for chrY"
#&gt; [1] "2021-08-24 21:50:52 - Generating ERs for chrM"
#&gt; [1] "2021-08-24 21:50:54 - Loading in GTF..."
#&gt; [1] "2021-08-24 21:51:28 - Obtaining non-overlapping exons"
#&gt; [1] "2021-08-24 21:51:29 - Calculating delta for ERs..."
#&gt; [1] "2021-08-24 21:51:39 - Obtaining optimal set of ERs..."</div><div class='input'>
<span class='va'>opt_ers</span>
</div><div class='output co'>#&gt; $opt_ers
#&gt; GRanges object with 345380 ranges and 0 metadata columns:
#&gt;            seqnames            ranges strand
#&gt;               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;        [1]     chr1       14468-14829      +
#&gt;        [2]     chr1       14970-14976      +
#&gt;        [3]     chr1       15029-15038      +
#&gt;        [4]     chr1       15796-15809      +
#&gt;        [5]     chr1       15934-15947      +
#&gt;        ...      ...               ...    ...
#&gt;   [345376]     chrY 10192024-10192091      -
#&gt;   [345377]     chrY 13286661-13286767      -
#&gt;   [345378]     chrY 13286878-13286959      -
#&gt;   [345379]     chrY 13287222-13287350      -
#&gt;   [345380]     chrM           1-16569      -
#&gt;   -------
#&gt;   seqinfo: 25 sequences from an unspecified genome; no seqlengths
#&gt; 
#&gt; $opt_mcc_mrg
#&gt; [1] "mcc_10" "mrg_20"
#&gt; 
#&gt; $deltas
#&gt; <span style='color: #555555;'># A tibble: 4 x 7</span><span>
#&gt;     mcc   mrg      sum  mean median n_eq_0 propor_eq_0
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>    </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>  </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>       </span><span style='color: #555555;font-style: italic;'>&lt;dbl&gt;</span><span>
#&gt; </span><span style='color: #555555;'>1</span><span>     5    10 42</span><span style='text-decoration: underline;'>342</span><span>549  796.     79  </span><span style='text-decoration: underline;'>19</span><span>014       0.357
#&gt; </span><span style='color: #555555;'>2</span><span>     5    20 33</span><span style='text-decoration: underline;'>095</span><span>527  686.     39  </span><span style='text-decoration: underline;'>19</span><span>267       0.399
#&gt; </span><span style='color: #555555;'>3</span><span>    10    10 33</span><span style='text-decoration: underline;'>189</span><span>459  801.     60  </span><span style='text-decoration: underline;'>16</span><span>668       0.402
#&gt; </span><span style='color: #555555;'>4</span><span>    10    20 25</span><span style='text-decoration: underline;'>021</span><span>958  667.     16  </span><span style='text-decoration: underline;'>16</span><span>897       0.450
#&gt; </div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Emmanuel Olagbaju.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


