<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>ODER workflow • ODER</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="ODER workflow">
<meta property="og:description" content="ODER">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ODER</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ODERflow.html">ODER workflow</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/eolagbaju/ODER/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ODERflow_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>ODER workflow</h1>
                        <h4 class="author">Emmanuel Olagbaju</h4>
            <address class="author_afil">
      UCL<br><a class="author_email" href="mailto:#"></a><a href="mailto:e.olagbaju@ucl.ac.uk" class="email">e.olagbaju@ucl.ac.uk</a>
      </address>
                  
            <h4 class="date">29 June 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/eolagbaju/ODER/blob/master/vignettes/ODERflow.Rmd"><code>vignettes/ODERflow.Rmd</code></a></small>
      <div class="hidden name"><code>ODERflow.Rmd</code></div>

    </div>

    
    
<div id="basics" class="section level1">
<h1 class="hasAnchor">
<a href="#basics" class="anchor"></a>Basics</h1>
<div id="install-oder" class="section level2">
<h2 class="hasAnchor">
<a href="#install-oder" class="anchor"></a>Install <code>ODER</code>
</h2>
<p><code>R</code> is an open-source statistical environment which can be easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.12/ODER">ODER</a></em> is a <code>R</code> package available via the <a href="http://bioconductor.org">Bioconductor</a> repository for packages. <code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/">CRAN</a> after which you can install <em><a href="https://bioconductor.org/packages/3.12/ODER">ODER</a></em> by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="op">}</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"ODER"</span><span class="op">)</span>

<span class="co">## Check that you have a valid Bioconductor installation</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html">valid</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div id="required-knowledge" class="section level2">
<h2 class="hasAnchor">
<a href="#required-knowledge" class="anchor"></a>Required knowledge</h2>
<p><em><a href="https://bioconductor.org/packages/3.12/ODER">ODER</a></em> is based on many other packages and in particular in those that have implemented the infrastructure needed for dealing with RNA-seq data (EDIT!). That is, packages like <em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em> (EDIT!).</p>
<p>If you are asking yourself the question “Where do I start using Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU">this blog post</a>.</p>
</div>
<div id="asking-for-help" class="section level2">
<h2 class="hasAnchor">
<a href="#asking-for-help" class="anchor"></a>Asking for help</h2>
<p>As package developers, we try to explain clearly how to use our packages and in which order to use the functions. But <code>R</code> and <code>Bioconductor</code> have a steep learning curve so it is critical to learn where to ask for help. The blog post quoted above mentions some but we would like to highlight the <a href="https://support.bioconductor.org/">Bioconductor support site</a> as the main resource for getting help: remember to use the <code>ODER</code> tag and check <a href="https://support.bioconductor.org/t/ODER/">the older posts</a>. Other alternatives are available such as creating GitHub issues and tweeting. However, please note that if you want to receive help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/">posting guidelines</a>. It is particularly critical that you provide a small reproducible example and your session information so package developers can track down the source of the error.</p>
</div>
<div id="citing-oder" class="section level2">
<h2 class="hasAnchor">
<a href="#citing-oder" class="anchor"></a>Citing <code>ODER</code>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.12/ODER">ODER</a></em> will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Citation info</span>
<span class="co"># citation("ODER")</span></code></pre></div>
</div>
</div>
<div id="quick-start-to-using-to-oder" class="section level1">
<h1 class="hasAnchor">
<a href="#quick-start-to-using-to-oder" class="anchor"></a>Quick start to using to <code>ODER</code>
</h1>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/eolagbaju/ODER">"ODER"</a></span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org">magrittr</a></span><span class="op">)</span>
<span class="va">gtex_metadata</span> <span class="op">&lt;-</span> <span class="fu">recount</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/recount/man/all_metadata.html">all_metadata</a></span><span class="op">(</span><span class="st">"gtex"</span><span class="op">)</span>
<span class="co">#&gt; Setting options('download.file.method.GEOquery'='auto')</span>
<span class="co">#&gt; Setting options('GEOquery.inmemory.gpl'=FALSE)</span>
<span class="co">#&gt; 2021-06-29 20:24:29 downloading the metadata to /tmp/RtmpgPzDlc/metadata_clean_gtex.Rdata</span>
<span class="co"># gtex_metadata[["bigwig_file"]][grepl("fibroblast",gtex_metadata[["bigwig_file"]])]</span>
<span class="co"># dplyr::filter(gtex_metadata, bigwig_file %in% "fibroblast")</span>
<span class="va">gtex_metadata</span> <span class="op">&lt;-</span> <span class="va">gtex_metadata</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">project</span> <span class="op">==</span> <span class="st">"SRP012682"</span><span class="op">)</span>

<span class="co"># obtain path to example bw on recount2</span>
<span class="va">url</span> <span class="op">&lt;-</span> <span class="fu">recount</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study</a></span><span class="op">(</span>
    project <span class="op">=</span> <span class="st">"SRP012682"</span>,
    type <span class="op">=</span> <span class="st">"samples"</span>,
    download <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span>

<span class="co"># grep("fibroblast",url)</span>
<span class="co"># bw_path &lt;- ODER:::.file_cache(url[1])</span>
<span class="co"># bw_path &lt;- ODER:::.file_cache(url[58])</span>
<span class="co"># 84  91   98  106  141  217  245  253  280  301  306  310  318  328  330  375  396  488  522  547  561  588  599</span>
<span class="va">bw_path</span> <span class="op">&lt;-</span> <span class="fu">ODER</span><span class="fu">:::</span><span class="fu">.file_cache</span><span class="op">(</span><span class="va">url</span><span class="op">[</span><span class="fl">280</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># getting the mean coverage of the expressed regions</span>
<span class="va">coverage</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_coverage.html">get_coverage</a></span><span class="op">(</span>
    bw_paths <span class="op">=</span> <span class="va">bw_path</span>,
    <span class="co"># auc_raw = gtex_metadata[["auc"]][1],</span>
    <span class="co"># auc_raw = gtex_metadata[["auc"]][58],</span>
    auc_raw <span class="op">=</span> <span class="va">gtex_metadata</span><span class="op">[[</span><span class="st">"auc"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">280</span><span class="op">]</span>,
    auc_target <span class="op">=</span> <span class="fl">40e6</span> <span class="op">*</span> <span class="fl">100</span>, <span class="co"># target 40 million coverage with 100 bp length reads</span>
    chrs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chr21"</span>, <span class="st">"chr22"</span><span class="op">)</span>
<span class="op">)</span> <span class="co"># defaults to chr1-22, chrX, chrY, chrM if chrs = ""</span>
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:parallel':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
<span class="co">#&gt;     clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
<span class="co">#&gt;     parLapplyLB, parRapply, parSapply, parSapplyLB</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class="co">#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class="co">#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class="co">#&gt;     union, unique, unsplit, which.max, which.min</span>
<span class="co">#&gt; Loading required package: S4Vectors</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid</span>
<span class="co">#&gt; [1] "2021-06-29 20:24:42 - Obtaining mean coverage across 1 samples"</span>
<span class="co">#&gt; [1] "2021-06-29 20:24:42 - chr21"</span>
<span class="co">#&gt; [1] "2021-06-29 20:24:44 - chr22"</span>

<span class="va">coverage</span>
<span class="co">#&gt; $chr21</span>
<span class="co">#&gt; $chr21$meanCoverage</span>
<span class="co">#&gt; numeric-Rle of length 46709983 with 440321 runs</span>
<span class="co">#&gt;   Lengths:  5011530       76     4585       76 ...        2       75    13220</span>
<span class="co">#&gt;   Values : 0.000000 0.789416 0.000000 0.789416 ... 0.000000 0.789416 0.000000</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $chr22</span>
<span class="co">#&gt; $chr22$meanCoverage</span>
<span class="co">#&gt; numeric-Rle of length 50818468 with 766223 runs</span>
<span class="co">#&gt;   Lengths:  10537100         6        29 ...         4         1     13487</span>
<span class="co">#&gt;   Values :  0.000000  0.789416  1.578831 ...  1.578831  0.789416  0.000000</span>

<span class="va">ers</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ers.html">get_ers</a></span><span class="op">(</span>coverage <span class="op">=</span> <span class="va">coverage</span>, mccs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">6</span>, <span class="fl">8</span>, <span class="fl">10</span><span class="op">)</span>, mrgs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">20</span>, <span class="fl">30</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "2021-06-29 20:24:45 - Generating ERs for chr21"</span>
<span class="co">#&gt; [1] "2021-06-29 20:24:51 - Generating ERs for chr22"</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># download the gtf file from ensembl</span>
<span class="va">gtf_url</span> <span class="op">&lt;-</span> <span class="st">"http://ftp.ensembl.org/pub/release-103/gtf/homo_sapiens/Homo_sapiens.GRCh38.103.chr.gtf.gz"</span>
<span class="va">gtf_path</span> <span class="op">&lt;-</span> <span class="fu">ODER</span><span class="fu">:::</span><span class="fu">.file_cache</span><span class="op">(</span><span class="va">gtf_url</span><span class="op">)</span>

<span class="va">exons_no_overlap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_exons.html">get_exons</a></span><span class="op">(</span>gtf <span class="op">=</span> <span class="va">gtf_path</span>, ucsc_chr <span class="op">=</span> <span class="cn">TRUE</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; [1] "2021-06-29 20:24:55 - Loading in GTF..."</span>
<span class="co">#&gt; [1] "2021-06-29 20:25:33 - Obtaining non-overlapping exons"</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get the ers_delta</span>
<span class="va">ers_delta</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ers_delta.html">get_ers_delta</a></span><span class="op">(</span>ers <span class="op">=</span> <span class="va">ers</span>, opt_exons <span class="op">=</span> <span class="va">exons_no_overlap</span><span class="op">)</span>
<span class="co">#&gt; [1] "2021-06-29 20:25:35 - Calculating delta for ERs..."</span>
<span class="co"># if you have your own delta function you can enter it as an argument i.e.</span>
<span class="co"># delta_fun &lt;- your_delta_function</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get the opt_ers</span>
<span class="va">opt_ers</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_opt_ers.html">get_opt_ers</a></span><span class="op">(</span>ers <span class="op">=</span> <span class="va">ers</span>, ers_delta <span class="op">=</span> <span class="va">ers_delta</span><span class="op">)</span>
<span class="co">#&gt; [1] "2021-06-29 20:25:37 - Obtaining optimal set of ERs..."</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># visualise the spread of mccs and mrgs</span>
<span class="va">opt_plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_ers.html">plot_ers</a></span><span class="op">(</span>ers_delta <span class="op">=</span> <span class="va">ers_delta</span>, opt_mcc_mrg <span class="op">=</span> <span class="va">opt_ers</span><span class="op">[[</span><span class="st">"opt_mcc_mrg"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>

<span class="va">opt_plots</span></code></pre></div>
<p><img src="ODERflow_files/figure-html/plot_ers-1.png" width="700"></p>
<pre><code>#&gt; [1] "2021-06-29 20:25:38 - Obtaining co-ordinates of annotated exons and junctions from gtf/gff3..."
#&gt; [1] "2021-06-29 20:25:38 - Importing gtf/gff3 as a TxDb..."
#&gt; Import genomic features from the file as a GRanges object ... OK
#&gt; Prepare the 'metadata' data frame ... OK
#&gt; Make the TxDb object ...
#&gt; Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The "phase" metadata column contains non-NA values for features of type
#&gt;   stop_codon. This information was ignored.
#&gt; OK
#&gt; [1] "2021-06-29 20:26:58 - Getting junction annotation using overlapping exons..."
#&gt; [1] "2021-06-29 20:27:00 - Tidying junction annotation..."
#&gt; [1] "2021-06-29 20:27:01 - Deriving junction categories..."
#&gt; [1] "2021-06-29 20:27:04 - done!"
#&gt; [1] "2021-06-29 20:27:04 - Finding junctions overlapping ers..."
#&gt; [1] "2021-06-29 20:27:36 - Generating a genomic state..."
#&gt; Import genomic features from the file as a GRanges object ... OK
#&gt; Prepare the 'metadata' data frame ... OK
#&gt; Make the TxDb object ...
#&gt; Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The "phase" metadata column contains non-NA values for features of type
#&gt;   stop_codon. This information was ignored.
#&gt; OK
#&gt; Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The "phase" metadata column contains non-NA values for features of type
#&gt;   stop_codon. This information was ignored.
#&gt; extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens
#&gt; 'select()' returned 1:1 mapping between keys and columns
#&gt; 2021-06-29 20:29:22 annotateRegions: counting
#&gt; 2021-06-29 20:29:22 annotateRegions: annotating
#&gt; GRanges object with 6513 ranges and 4 metadata columns:
#&gt;          seqnames            ranges strand |
#&gt;             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |
#&gt;      [1]    chr21   5116348-5117227      * |
#&gt;      [2]    chr21   5118695-5118847      * |
#&gt;      [3]    chr21   5120788-5121008      * |
#&gt;      [4]    chr21   5121572-5121598      * |
#&gt;      [5]    chr21   5121711-5121803      * |
#&gt;      ...      ...               ...    ... .
#&gt;   [6509]    chr22 50802977-50802983      * |
#&gt;   [6510]    chr22 50803456-50803599      * |
#&gt;   [6511]    chr22 50804134-50804214      * |
#&gt;   [6512]    chr22 50804499-50804528      * |
#&gt;   [6513]    chr22 50804709-50804806      * |
#&gt;                                                                                    grl
#&gt;                                                                          &lt;GRangesList&gt;
#&gt;      [1]   chr21:5026423-5323718:+,chr21:5075464-5140960:+,chr21:5105364-5142531:-,...
#&gt;      [2]   chr21:5026423-5323718:+,chr21:5075464-5140960:+,chr21:5105364-5142531:-,...
#&gt;      [3]   chr21:5026423-5323718:+,chr21:5075464-5140960:+,chr21:5105364-5142531:-,...
#&gt;      [4]   chr21:5026423-5323718:+,chr21:5075464-5140960:+,chr21:5105364-5142531:-,...
#&gt;      [5]   chr21:5026423-5323718:+,chr21:5075464-5140960:+,chr21:5105364-5142531:-,...
#&gt;      ...                                                                           ...
#&gt;   [6509] chr22:50363416-50804892:-,chr22:50460674-50807396:+,chr22:50770378-50805745:-
#&gt;   [6510] chr22:50363416-50804892:-,chr22:50460674-50807396:+,chr22:50770378-50805745:-
#&gt;   [6511] chr22:50363416-50804892:-,chr22:50460674-50807396:+,chr22:50770378-50805745:-
#&gt;   [6512] chr22:50363416-50804892:-,chr22:50460674-50807396:+,chr22:50770378-50805745:-
#&gt;   [6513] chr22:50363416-50804892:-,chr22:50460674-50807396:+,chr22:50770378-50805745:-
#&gt;                                                    genes  annotation  og_index
#&gt;                                          &lt;CharacterList&gt; &lt;character&gt; &lt;integer&gt;
#&gt;      [1] ENSG00000275464,ENSG00000280071,ENSG00000276612        exon         1
#&gt;      [2] ENSG00000275464,ENSG00000280071,ENSG00000276612        exon         2
#&gt;      [3] ENSG00000275464,ENSG00000280071,ENSG00000276612        exon         3
#&gt;      [4] ENSG00000275464,ENSG00000280071,ENSG00000276612        exon         4
#&gt;      [5] ENSG00000275464,ENSG00000280071,ENSG00000276612        exon         5
#&gt;      ...                                             ...         ...       ...
#&gt;   [6509]                                                  intergenic      6509
#&gt;   [6510]                                                  intergenic      6510
#&gt;   [6511]                                                  intergenic      6511
#&gt;   [6512]                                                  intergenic      6512
#&gt;   [6513]                                                  intergenic      6513
#&gt;   -------
#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<pre><code>#&gt; [1] "2021-06-29 20:29:23 - Generating a genomic state..."
#&gt; Import genomic features from the file as a GRanges object ... OK
#&gt; Prepare the 'metadata' data frame ... OK
#&gt; Make the TxDb object ...
#&gt; Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The "phase" metadata column contains non-NA values for features of type
#&gt;   stop_codon. This information was ignored.
#&gt; OK
#&gt; Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The "phase" metadata column contains non-NA values for features of type
#&gt;   stop_codon. This information was ignored.
#&gt; extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens
#&gt; 'select()' returned 1:1 mapping between keys and columns</code></pre>
<pre><code>#&gt; [1] "2021-06-29 20:31:08 - Obtaining co-ordinates of annotated exons and junctions from gtf/gff3..."
#&gt; [1] "2021-06-29 20:31:08 - Importing gtf/gff3 as a TxDb..."
#&gt; Import genomic features from the file as a GRanges object ... OK
#&gt; Prepare the 'metadata' data frame ... OK
#&gt; Make the TxDb object ...
#&gt; Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The "phase" metadata column contains non-NA values for features of type
#&gt;   stop_codon. This information was ignored.
#&gt; OK
#&gt; [1] "2021-06-29 20:32:21 - Getting junction annotation using overlapping exons..."
#&gt; [1] "2021-06-29 20:32:23 - Tidying junction annotation..."
#&gt; [1] "2021-06-29 20:32:23 - Deriving junction categories..."
#&gt; [1] "2021-06-29 20:32:26 - done!"
#&gt; [1] "2021-06-29 20:32:26 - Finding junctions overlapping ers..."
#&gt; [1] "2021-06-29 20:32:53 - Generating a genomic state..."
#&gt; Import genomic features from the file as a GRanges object ... OK
#&gt; Prepare the 'metadata' data frame ... OK
#&gt; Make the TxDb object ...
#&gt; Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The "phase" metadata column contains non-NA values for features of type
#&gt;   stop_codon. This information was ignored.
#&gt; OK
#&gt; Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The "phase" metadata column contains non-NA values for features of type
#&gt;   stop_codon. This information was ignored.
#&gt; extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens
#&gt; 'select()' returned 1:1 mapping between keys and columns
#&gt; [1] "2021-06-29 20:34:40 - Annotating the Expressed regions..."
#&gt; 2021-06-29 20:34:40 annotateRegions: counting
#&gt; 2021-06-29 20:34:40 annotateRegions: annotating
#&gt; [1] "2021-06-29 20:35:19 - done!"</code></pre>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.12/biocthis">biocthis</a></em>.</p>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Create the vignette</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown">"rmarkdown"</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span><span class="op">(</span><span class="st">"ODER.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span><span class="op">)</span><span class="op">)</span>

<span class="co">## Extract the R code</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/">"knitr"</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html">knit</a></span><span class="op">(</span><span class="st">"ODER.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code>#&gt; [1] "2021-06-29 20:35:20 UTC"</code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code>#&gt; Time difference of 11.077 mins</code></pre>
<p><code>R</code> session information.</p>
<pre><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  setting  value                       
#&gt;  version  R version 4.0.3 (2020-10-10)
#&gt;  os       Ubuntu 20.04 LTS            
#&gt;  system   x86_64, linux-gnu           
#&gt;  ui       X11                         
#&gt;  language (EN)                        
#&gt;  collate  en_US.UTF-8                 
#&gt;  ctype    en_US.UTF-8                 
#&gt;  tz       Etc/UTC                     
#&gt;  date     2021-06-29                  
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  package              * version  date       lib source        
#&gt;  abind                  1.4-5    2016-07-21 [1] RSPM (R 4.0.3)
#&gt;  AnnotationDbi          1.52.0   2020-10-27 [1] Bioconductor  
#&gt;  askpass                1.1      2019-01-13 [2] RSPM (R 4.0.3)
#&gt;  assertthat             0.2.1    2019-03-21 [2] RSPM (R 4.0.3)
#&gt;  backports              1.2.1    2020-12-09 [1] RSPM (R 4.0.3)
#&gt;  base64enc              0.1-3    2015-07-28 [2] RSPM (R 4.0.3)
#&gt;  basilisk               1.2.1    2020-12-16 [1] Bioconductor  
#&gt;  basilisk.utils         1.2.2    2021-01-27 [1] Bioconductor  
#&gt;  Biobase                2.50.0   2020-10-27 [1] Bioconductor  
#&gt;  BiocFileCache          1.14.0   2020-10-27 [1] Bioconductor  
#&gt;  BiocGenerics         * 0.36.1   2021-04-16 [1] Bioconductor  
#&gt;  BiocManager            1.30.10  2019-11-16 [1] RSPM (R 4.0.0)
#&gt;  BiocParallel           1.24.1   2020-11-06 [1] Bioconductor  
#&gt;  BiocStyle            * 2.18.1   2020-11-24 [1] Bioconductor  
#&gt;  biomaRt                2.46.3   2021-02-09 [1] Bioconductor  
#&gt;  Biostrings             2.58.0   2020-10-27 [1] Bioconductor  
#&gt;  bit                    4.0.4    2020-08-04 [1] RSPM (R 4.0.3)
#&gt;  bit64                  4.0.5    2020-08-30 [1] RSPM (R 4.0.3)
#&gt;  bitops                 1.0-6    2013-08-17 [1] RSPM (R 4.0.3)
#&gt;  blob                   1.2.1    2020-01-20 [1] RSPM (R 4.0.3)
#&gt;  bookdown               0.21     2020-10-13 [1] RSPM (R 4.0.2)
#&gt;  broom                  0.7.4    2021-01-29 [1] RSPM (R 4.0.3)
#&gt;  BSgenome               1.58.0   2020-10-27 [1] Bioconductor  
#&gt;  bumphunter             1.32.0   2020-10-27 [1] Bioconductor  
#&gt;  cachem                 1.0.4    2021-02-13 [2] RSPM (R 4.0.3)
#&gt;  car                    3.0-10   2020-09-29 [1] RSPM (R 4.0.3)
#&gt;  carData                3.0-4    2020-05-22 [1] RSPM (R 4.0.3)
#&gt;  cellranger             1.1.0    2016-07-27 [1] RSPM (R 4.0.3)
#&gt;  checkmate              2.0.0    2020-02-06 [1] RSPM (R 4.0.3)
#&gt;  cli                    2.3.0    2021-01-31 [2] RSPM (R 4.0.3)
#&gt;  cluster                2.1.1    2021-02-14 [3] RSPM (R 4.0.3)
#&gt;  codetools              0.2-18   2020-11-04 [3] RSPM (R 4.0.3)
#&gt;  colorspace             2.0-0    2020-11-11 [1] RSPM (R 4.0.3)
#&gt;  cowplot                1.1.1    2020-12-30 [1] RSPM (R 4.0.3)
#&gt;  crayon                 1.4.1    2021-02-08 [2] RSPM (R 4.0.3)
#&gt;  curl                   4.3      2019-12-02 [2] RSPM (R 4.0.3)
#&gt;  dasper                 1.0.0    2020-10-27 [1] Bioconductor  
#&gt;  data.table             1.13.6   2020-12-30 [1] RSPM (R 4.0.3)
#&gt;  DBI                    1.1.1    2021-01-15 [1] RSPM (R 4.0.3)
#&gt;  dbplyr                 2.1.0    2021-02-03 [1] RSPM (R 4.0.3)
#&gt;  DelayedArray           0.16.3   2021-03-24 [1] Bioconductor  
#&gt;  derfinder              1.24.2   2020-12-18 [1] Bioconductor  
#&gt;  derfinderHelper        1.24.1   2020-12-18 [1] Bioconductor  
#&gt;  desc                   1.2.0    2018-05-01 [2] RSPM (R 4.0.3)
#&gt;  digest                 0.6.27   2020-10-24 [2] RSPM (R 4.0.3)
#&gt;  doRNG                  1.8.2    2020-01-27 [1] RSPM (R 4.0.3)
#&gt;  downloader             0.4      2015-07-09 [1] RSPM (R 4.0.3)
#&gt;  dplyr                  1.0.4    2021-02-02 [1] RSPM (R 4.0.3)
#&gt;  ellipsis               0.3.1    2020-05-15 [2] RSPM (R 4.0.3)
#&gt;  evaluate               0.14     2019-05-28 [2] RSPM (R 4.0.3)
#&gt;  farver                 2.0.3    2020-01-16 [1] RSPM (R 4.0.3)
#&gt;  fastmap                1.1.0    2021-01-25 [2] RSPM (R 4.0.3)
#&gt;  filelock               1.0.2    2018-10-05 [1] RSPM (R 4.0.0)
#&gt;  forcats                0.5.1    2021-01-27 [1] RSPM (R 4.0.3)
#&gt;  foreach                1.5.1    2020-10-15 [1] RSPM (R 4.0.3)
#&gt;  foreign                0.8-81   2020-12-22 [3] RSPM (R 4.0.3)
#&gt;  Formula                1.2-4    2020-10-16 [1] RSPM (R 4.0.3)
#&gt;  fs                     1.5.0    2020-07-31 [2] RSPM (R 4.0.3)
#&gt;  generics               0.1.0    2020-10-31 [1] RSPM (R 4.0.3)
#&gt;  GenomeInfoDb         * 1.26.7   2021-04-08 [1] Bioconductor  
#&gt;  GenomeInfoDbData       1.2.4    2021-06-28 [1] Bioconductor  
#&gt;  GenomicAlignments      1.26.0   2020-10-27 [1] Bioconductor  
#&gt;  GenomicFeatures        1.42.3   2021-04-01 [1] Bioconductor  
#&gt;  GenomicFiles           1.26.0   2020-10-27 [1] Bioconductor  
#&gt;  GenomicRanges          1.42.0   2020-10-27 [1] Bioconductor  
#&gt;  GEOquery               2.58.0   2020-10-27 [1] Bioconductor  
#&gt;  ggplot2                3.3.3    2020-12-30 [1] RSPM (R 4.0.3)
#&gt;  ggpubr                 0.4.0    2020-06-27 [1] RSPM (R 4.0.3)
#&gt;  ggrepel                0.9.1    2021-01-15 [1] RSPM (R 4.0.3)
#&gt;  ggsignif               0.6.0    2019-08-08 [1] RSPM (R 4.0.3)
#&gt;  glue                   1.4.2    2020-08-27 [2] RSPM (R 4.0.3)
#&gt;  gridExtra              2.3      2017-09-09 [1] RSPM (R 4.0.3)
#&gt;  gtable                 0.3.0    2019-03-25 [1] RSPM (R 4.0.3)
#&gt;  haven                  2.3.1    2020-06-01 [1] RSPM (R 4.0.3)
#&gt;  highr                  0.8      2019-03-20 [2] RSPM (R 4.0.3)
#&gt;  Hmisc                  4.4-2    2020-11-29 [1] RSPM (R 4.0.3)
#&gt;  hms                    1.0.0    2021-01-13 [1] RSPM (R 4.0.3)
#&gt;  htmlTable              2.1.0    2020-09-16 [1] RSPM (R 4.0.3)
#&gt;  htmltools              0.5.1.1  2021-01-22 [2] RSPM (R 4.0.3)
#&gt;  htmlwidgets            1.5.3    2020-12-10 [2] RSPM (R 4.0.3)
#&gt;  httr                   1.4.2    2020-07-20 [2] RSPM (R 4.0.3)
#&gt;  IRanges              * 2.24.1   2020-12-12 [1] Bioconductor  
#&gt;  iterators              1.0.13   2020-10-15 [1] RSPM (R 4.0.3)
#&gt;  jpeg                   0.1-8.1  2019-10-24 [1] RSPM (R 4.0.3)
#&gt;  jsonlite               1.7.2    2020-12-09 [2] RSPM (R 4.0.3)
#&gt;  knitr                  1.31     2021-01-27 [2] RSPM (R 4.0.3)
#&gt;  labeling               0.4.2    2020-10-20 [1] RSPM (R 4.0.3)
#&gt;  lattice                0.20-41  2020-04-02 [3] CRAN (R 4.0.3)
#&gt;  latticeExtra           0.6-29   2019-12-19 [1] RSPM (R 4.0.3)
#&gt;  lifecycle              1.0.0    2021-02-15 [2] RSPM (R 4.0.3)
#&gt;  limma                  3.46.0   2020-10-27 [1] Bioconductor  
#&gt;  locfit                 1.5-9.4  2020-03-25 [1] RSPM (R 4.0.3)
#&gt;  lubridate              1.7.9.2  2020-11-13 [1] RSPM (R 4.0.3)
#&gt;  magrittr             * 2.0.1    2020-11-17 [2] RSPM (R 4.0.3)
#&gt;  Matrix                 1.3-2    2021-01-06 [3] RSPM (R 4.0.3)
#&gt;  MatrixGenerics         1.2.1    2021-01-30 [1] Bioconductor  
#&gt;  matrixStats            0.58.0   2021-01-29 [1] RSPM (R 4.0.3)
#&gt;  memoise                2.0.0    2021-01-26 [2] RSPM (R 4.0.3)
#&gt;  munsell                0.5.0    2018-06-12 [1] RSPM (R 4.0.3)
#&gt;  nnet                   7.3-15   2021-01-24 [3] RSPM (R 4.0.3)
#&gt;  ODER                 * 0.99.8   2021-06-29 [1] Bioconductor  
#&gt;  openssl                1.4.3    2020-09-18 [2] RSPM (R 4.0.3)
#&gt;  openxlsx               4.2.3    2020-10-27 [1] RSPM (R 4.0.3)
#&gt;  pillar                 1.4.7    2020-11-20 [2] RSPM (R 4.0.3)
#&gt;  pkgconfig              2.0.3    2019-09-22 [2] RSPM (R 4.0.3)
#&gt;  pkgdown                1.6.1    2020-09-12 [1] RSPM (R 4.0.2)
#&gt;  plyr                   1.8.6    2020-03-03 [1] RSPM (R 4.0.3)
#&gt;  png                    0.1-7    2013-12-03 [1] RSPM (R 4.0.3)
#&gt;  prettyunits            1.1.1    2020-01-24 [2] RSPM (R 4.0.3)
#&gt;  progress               1.2.2    2019-05-16 [1] RSPM (R 4.0.3)
#&gt;  purrr                  0.3.4    2020-04-17 [2] RSPM (R 4.0.3)
#&gt;  qvalue                 2.22.0   2020-10-27 [1] Bioconductor  
#&gt;  R6                     2.5.0    2020-10-28 [2] RSPM (R 4.0.3)
#&gt;  ragg                   1.1.0    2021-02-15 [1] RSPM (R 4.0.3)
#&gt;  rappdirs               0.3.3    2021-01-31 [2] RSPM (R 4.0.3)
#&gt;  RColorBrewer           1.1-2    2014-12-07 [1] RSPM (R 4.0.3)
#&gt;  Rcpp                   1.0.6    2021-01-15 [2] RSPM (R 4.0.3)
#&gt;  RCurl                  1.98-1.2 2020-04-18 [1] RSPM (R 4.0.3)
#&gt;  readr                  1.4.0    2020-10-05 [1] RSPM (R 4.0.3)
#&gt;  readxl                 1.3.1    2019-03-13 [1] RSPM (R 4.0.3)
#&gt;  recount                1.16.1   2020-12-18 [1] Bioconductor  
#&gt;  RefManageR           * 1.3.0    2020-11-13 [1] RSPM (R 4.0.3)
#&gt;  rentrez                1.2.3    2020-11-10 [1] RSPM (R 4.0.3)
#&gt;  reshape2               1.4.4    2020-04-09 [1] RSPM (R 4.0.3)
#&gt;  reticulate             1.18     2020-10-25 [1] RSPM (R 4.0.3)
#&gt;  rio                    0.5.16   2018-11-26 [1] RSPM (R 4.0.3)
#&gt;  rlang                  0.4.10   2020-12-30 [2] RSPM (R 4.0.3)
#&gt;  rmarkdown              2.6      2020-12-14 [1] RSPM (R 4.0.3)
#&gt;  rngtools               1.5      2020-01-23 [1] RSPM (R 4.0.3)
#&gt;  rpart                  4.1-15   2019-04-12 [3] CRAN (R 4.0.3)
#&gt;  rprojroot              2.0.2    2020-11-15 [2] RSPM (R 4.0.3)
#&gt;  Rsamtools              2.6.0    2020-10-27 [1] Bioconductor  
#&gt;  RSQLite                2.2.3    2021-01-24 [1] RSPM (R 4.0.3)
#&gt;  rstatix                0.7.0    2021-02-13 [1] RSPM (R 4.0.3)
#&gt;  rstudioapi             0.13     2020-11-12 [2] RSPM (R 4.0.3)
#&gt;  rtracklayer            1.50.0   2020-10-27 [1] Bioconductor  
#&gt;  S4Vectors            * 0.28.1   2020-12-09 [1] Bioconductor  
#&gt;  scales                 1.1.1    2020-05-11 [1] RSPM (R 4.0.3)
#&gt;  sessioninfo          * 1.1.1    2018-11-05 [2] RSPM (R 4.0.3)
#&gt;  stringi                1.5.3    2020-09-09 [2] RSPM (R 4.0.3)
#&gt;  stringr                1.4.0    2019-02-10 [2] RSPM (R 4.0.3)
#&gt;  SummarizedExperiment   1.20.0   2020-10-27 [1] Bioconductor  
#&gt;  survival               3.2-7    2020-09-28 [3] CRAN (R 4.0.3)
#&gt;  systemfonts            1.0.1    2021-02-09 [1] RSPM (R 4.0.3)
#&gt;  textshaping            0.3.0    2021-02-10 [1] RSPM (R 4.0.3)
#&gt;  tibble                 3.0.6    2021-01-29 [2] RSPM (R 4.0.3)
#&gt;  tidyr                  1.1.2    2020-08-27 [1] RSPM (R 4.0.3)
#&gt;  tidyselect             1.1.0    2020-05-11 [1] RSPM (R 4.0.3)
#&gt;  VariantAnnotation      1.36.0   2020-10-27 [1] Bioconductor  
#&gt;  vctrs                  0.3.6    2020-12-17 [2] RSPM (R 4.0.3)
#&gt;  withr                  2.4.1    2021-01-26 [2] RSPM (R 4.0.3)
#&gt;  xfun                   0.21     2021-02-10 [2] RSPM (R 4.0.3)
#&gt;  XML                    3.99-0.5 2020-07-23 [1] RSPM (R 4.0.3)
#&gt;  xml2                   1.3.2    2020-04-23 [2] RSPM (R 4.0.3)
#&gt;  XVector                0.30.0   2020-10-27 [1] Bioconductor  
#&gt;  yaml                   2.2.1    2020-02-01 [2] RSPM (R 4.0.3)
#&gt;  zip                    2.1.1    2020-08-27 [2] RSPM (R 4.0.3)
#&gt;  zlibbioc               1.36.0   2020-10-27 [1] Bioconductor  
#&gt; 
#&gt; [1] /__w/_temp/Library
#&gt; [2] /usr/local/lib/R/site-library
#&gt; [3] /usr/local/lib/R/library</code></pre>
</div>
<div id="bibliography" class="section level1">
<h1 class="hasAnchor">
<a href="#bibliography" class="anchor"></a>Bibliography</h1>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Emmanuel Olagbaju.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
